# mayhem_program::instructions::pump_interface_v2::buy_v2::handler  (yours 9408 B at .text+0x1fc88; original 18648 B at .text+0x223c0)
# distance 2687 instructions; evidence: called-by mayhem_program::__private::__global::buy_v2#a800; same-callees 19/24
--- original
+++ yours
@@ -1,304 +1,357 @@
-mov64 r7, r5
-mov64 r8, r4
-stxdw [r10-1848], r3
-stxdw [r10-1840], r2
+stxdw [r10-3336], r5
+mov64 r9, r4
+stxdw [r10-3312], r3
+stxdw [r10-3320], r2
+ldxdw r8, [r1+8]
+ldxdw r2, [r8+32]
+ldxdw r4, [r2]
+ldxdw r2, [r4]
+mov64 r3, 1
+stxdw [r10-3328], r3
+lddw r3, 0x8481abfe57889b06
+mov64 r7, 1
+jeq r2, r3, +0x78
+mov64 r7, 0
+ldxdw r3, [r4+8]
+lddw r0, 0x35c01846637f68fb
+mov64 r5, 1
+jeq r3, r0, +0xa8
+mov64 r5, 0
+and64 r7, r5
+ldxdw r5, [r4+16]
+lddw r6, 0x553beb1adc39c4da
+mov64 r0, 1
+jeq r5, r6, +0xe0
+mov64 r0, 0
+and64 r7, r0
+ldxdw r4, [r4+24]
+lddw r6, 0x100000000f0a098
+mov64 r0, 1
+jeq r4, r6, +0x118
+mov64 r0, 0
+and64 r7, r0
+and64 r7, 1
+jne r7, 0, +0x150
+jne r2, 0, +0x1f8
+jne r3, 0, +0x1f8
+jne r5, 0, +0x1f8
+jne r4, 0, +0x1f8
+ldxdw r2, [r8+8]
+ldxb r2, [r2+253]
+jeq r2, 0, +0x220
+mov64 r1, r10
+add64 r1, -552
+lddw r2, .rodata[00000000]
+call mayhem_program::MayhemError::name#2984
+mov64 r1, r10
+add64 r1, -528
+mov64 r2, 0
+call <T as alloc::string::SpecToString>::spec_to_string#2731
+lddw r1, .rodata[70726f6772616d73]
+stxdw [r10-576], r1
+stb [r10-504], 2
+stw [r10-560], 76
+stdw [r10-568], 61
+stdw [r10-584], 0
+stw [r10-432], 6000
+ja +0x588
+mov64 r2, 0
+stxdw [r10-3328], r2
+ldxdw r2, [r8+8]
+ldxb r2, [r2+253]
+jne r2, 0, +0x168
+stxdw [r10-3344], r1
+mov64 r1, r10
+add64 r1, -584
+call solana_sysvar::clock::<impl solana_sysvar::Sysvar for solana_clock::Clock>::get#06ec
+ldxw r1, [r10-584]
+jne r1, 0, +0x2300
+ldxdw r1, [r10-544]
+ldxdw r2, [r8+16]
+ldxdw r3, [r2+88]
+jeq r3, 0, +0x370
+ldxdw r4, [r8+8]
+ldxdw r6, [r4+237]
+jeq r6, 0, +0x370
+mov64 r0, 1
+mov64 r4, 1
+jslt r6, 0, +0x4b0
+mov64 r4, 0
+mov64 r5, r3
+add64 r5, r6
+jsge r5, r3, +0x4c8
+xor64 r4, r0
+and64 r4, 1
+jne r4, 0, +0x4e8
+jsge r1, r5, +0x370
+mov64 r1, r10
+add64 r1, -552
+lddw r2, .rodata[0f000000]
+call mayhem_program::instructions::pump_interface::sell::SellError::name#46b1
+mov64 r1, r10
+add64 r1, -528
+mov64 r2, 15
+call <T as alloc::string::SpecToString>::spec_to_string#d382
+lddw r1, .rodata[70726f6772616d73]
+stxdw [r10-576], r1
+stb [r10-504], 2
+stw [r10-560], 80
+stdw [r10-568], 61
+stdw [r10-584], 0
+stw [r10-432], 6015
+ja +0x588
+stxdw [r2+88], r1
+mov64 r1, r10
+add64 r1, -584
+call solana_sysvar::clock::<impl solana_sysvar::Sysvar for solana_clock::Clock>::get#06ec
+ldxw r1, [r10-584]
+jne r1, 0, +0x2360
+ldxdw r2, [r10-544]
+ldxdw r1, [r8+16]
+ldxdw r3, [r1+80]
+jsge r2, r3, +0x500
+ldxb r2, [r1+104]
+jeq r2, 1, +0x5b0
+mov64 r2, 144
+ldxdw r3, [r10-3328]
+jne r3, 0, +0x3f0
+mov64 r2, 192
+ldxdw r3, [r8+8]
+add64 r3, r2
+ldxdw r2, [r3]
+ldxdw r3, [r8+104]
+ldxdw r3, [r3+96]
+jge r3, r2, +0x5a8
+mov64 r1, r10
+add64 r1, -552
+lddw r2, .rodata[030000]
+call mayhem_program::instructions::pump_interface::sell::SellError::name#46b1
+mov64 r1, r10
+add64 r1, -528
+mov64 r2, 3
+call <T as alloc::string::SpecToString>::spec_to_string#d382
+lddw r1, .rodata[70726f6772616d73]
+stxdw [r10-576], r1
+stb [r10-504], 2
+stw [r10-560], 84
+stdw [r10-568], 61
+stdw [r10-584], 0
+stw [r10-432], 6003
+ja +0x588
+mov64 r5, r3
+add64 r5, r6
+jslt r5, r3, +0x2c0
+mov64 r0, 0
+xor64 r4, r0
+and64 r4, 1
+jeq r4, 0, +0x2d8
+lddw r1, panic_location(programs/mayhem-program/src/instructions/pump_interface_v2.rs:80:41)
+call core::panicking::panic_const::panic_const_add_overflow#3493
+mov64 r1, r10
+add64 r1, -552
+lddw r2, .rodata[04000000]
+call mayhem_program::instructions::pump_interface::sell::SellError::name#46b1
+mov64 r1, r10
+add64 r1, -528
+mov64 r2, 4
+call <T as alloc::string::SpecToString>::spec_to_string#d382
+lddw r1, .rodata[70726f6772616d73]
+stxdw [r10-576], r1
+stb [r10-504], 2
+stw [r10-560], 82
+stdw [r10-568], 61
+stdw [r10-584], 0
+stw [r10-432], 6004
+mov64 r1, r10
+add64 r1, -584
+call <anchor_lang::error::Error as core::convert::From<anchor_lang::error::AnchorError>>::from#5ca3
+exit
+stb [r1+104], 1
+ldxdw r1, [r8+104]
+ldxdw r2, [r1+72]
+stxdw [r10-3352], r2
+jeq r2, 0, +0x1738
+stxdw [r10-3368], r8
+stxdw [r10-3376], r1
+ldxdw r1, [r1+80]
+stxdw [r10-3360], r1
+and64 r9, 65535
+ldxdw r1, [r10-3312]
+mov64 r2, 0
+mov64 r3, r9
+mov64 r4, 0
+call __multi3
+mov64 r6, r0
+mov64 r8, r1
+ldxdw r1, [r10-3320]
+mov64 r2, 0
+mov64 r3, r9
+mov64 r4, 0
+call __multi3
+mov64 r2, r1
+add64 r2, r6
+mov64 r4, 1
+mov64 r3, 1
+jne r8, 0, +0x6a8
+mov64 r3, 0
+jge r2, r1, +0x6b0
+or64 r3, r4
+jne r3, 1, +0x6c8
+ja +0x1738
+jlt r2, r1, +0x690
+mov64 r4, 0
+or64 r3, r4
+jeq r3, 1, +0x1738
+mov64 r1, r0
+mov64 r3, 10000
+mov64 r4, 0
+call __udivti3
+mov64 r6, r0
 mov64 r9, r1
-ldxdw r6, [r9+8]
-ldxdw r2, [r6+8]
-ldxdw r1, [r6+16]
-add64 r1, 8
-add64 r2, 8
-call orig_fn_2dbf8
-jne r0, 2, +0x4b0
-ldxdw r1, [r6+104]
-ldxdw r2, [r1+96]
-ldxdw r3, [r6+8]
-ldxdw r1, [r6+16]
-ldxdw r4, [r6+32]
-ldxdw r4, [r4]
-ldxdw r5, [r4+24]
-stxdw [r10-1808], r5
-ldxdw r5, [r4+16]
-stxdw [r10-1816], r5
-ldxdw r5, [r4+8]
-stxdw [r10-1824], r5
-ldxdw r4, [r4]
-stxdw [r10-1832], r4
-add64 r1, 8
-add64 r3, 8
-mov64 r4, r10
-add64 r4, -1832
-call orig_fn_2e0a0
-jne r0, 2, +0x4b0
-ldxdw r1, [r6+104]
-stxdw [r10-1880], r1
-ldxdw r1, [r1+72]
-stxdw [r10-1872], r1
-jeq r1, 0, +0x4790
-stxdw [r10-1912], r9
-stxdw [r10-1896], r7
-stxdw [r10-1864], r6
-and64 r8, 65535
-ldxdw r1, [r10-1848]
-mov64 r2, 0
-mov64 r3, r8
-mov64 r4, 0
-call __multi3
-mov64 r7, r0
-mov64 r6, r1
-ldxdw r1, [r10-1840]
-mov64 r2, 0
-mov64 r3, r8
-mov64 r4, 0
-call __multi3
-mov64 r8, r0
-mov64 r9, r1
-add64 r9, r7
-mov64 r2, 1
-mov64 r7, 1
-jne r6, 0, +0x1e0
-mov64 r7, 0
-jlt r9, r1, +0x1f0
-mov64 r2, 0
-or64 r7, r2
-ldxdw r1, [r10-1880]
-ldxdw r1, [r1+80]
-stxdw [r10-1904], r1
+mov64 r1, 1
+mov64 r8, 1
+ldxdw r2, [r10-3320]
+jge r2, r6, +0x838
+mov64 r8, 0
+ldxdw r2, [r10-3312]
+jlt r2, r9, +0x848
+ldxdw r2, [r10-3312]
+jeq r2, r9, +0x748
+mov64 r8, r1
+ldxdw r1, [r10-3360]
 mov64 r2, 0
 lddw r3, 0x38d7ea4c68000
 mov64 r4, 0
 call __multi3
 mov64 r2, r1
 mov64 r1, r0
-ldxdw r3, [r10-1872]
+ldxdw r3, [r10-3352]
 mov64 r4, 0
 call __udivti3
-stxdw [r10-1888], r0
-stxdw [r10-1856], r1
-and64 r7, 1
-jne r7, 0, +0x4790
-mov64 r1, r8
-mov64 r2, r9
-mov64 r3, 10000
-mov64 r4, 0
-call __udivti3
+and64 r8, 1
+jeq r8, 0, +0x1738
+ldxdw r5, [r10-3320]
+mov64 r4, r5
+add64 r4, r6
 mov64 r3, 1
 mov64 r2, 1
-ldxdw r4, [r10-1840]
-jlt r4, r0, +0x318
+jlt r4, r5, +0x7e8
 mov64 r2, 0
-ldxdw r7, [r10-1864]
-ldxdw r8, [r10-1848]
-jge r8, r1, +0x330
-mov64 r9, 2
-jeq r8, r1, +0x348
+ldxdw r5, [r10-3312]
+mov64 r4, r5
+add64 r4, r9
+add64 r4, r2
+jlt r4, r5, +0x868
+mov64 r3, 0
+ldxdw r5, [r10-3312]
+jne r4, r5, +0x878
+jne r2, 1, +0x888
+ja +0x1738
+ldxdw r2, [r10-3312]
+jge r2, r9, +0x730
+mov64 r1, 0
+ldxdw r2, [r10-3312]
+jne r2, r9, +0x740
+ja +0x748
+ldxdw r5, [r10-3312]
+jeq r4, r5, +0x828
+mov64 r2, r3
+jeq r2, 1, +0x1738
+mov64 r3, r6
+mov64 r2, 1
+mov64 r4, 1
+ldxdw r5, [r10-3320]
+jlt r5, r3, +0x8b8
+mov64 r4, 0
+ldxdw r5, [r10-3312]
+sub64 r5, r9
+sub64 r5, r4
+ldxdw r8, [r10-3320]
+sub64 r8, r6
+mov64 r4, 1
+jlt r0, r8, +0x950
+mov64 r4, 0
+mov64 r8, 1
+jge r1, r5, +0x960
+jeq r1, r5, +0x970
+mov64 r4, r8
+ldxdw r5, [r10-3320]
+add64 r6, r5
+jge r6, r3, +0x988
+and64 r4, 1
+ldxdw r8, [r10-3368]
+jeq r4, 0, +0x9a8
+ja +0xa08
+mov64 r8, 1
+jlt r1, r5, +0x908
+mov64 r8, 0
+jne r1, r5, +0x910
+ldxdw r5, [r10-3320]
+add64 r6, r5
+jlt r6, r3, +0x930
+mov64 r2, 0
+and64 r4, 1
+ldxdw r8, [r10-3368]
+jne r4, 0, +0xa08
+ldxdw r3, [r10-3312]
+add64 r9, r3
+add64 r9, r2
+mov64 r3, 1
+mov64 r2, 1
+jle r0, r6, +0xa20
+mov64 r2, 0
+jgt r1, r9, +0xa28
+jeq r1, r9, +0xa38
 mov64 r2, r3
 and64 r2, 1
-jne r2, 0, +0x358
-ja +0x4b8
-ldxdw r7, [r10-1864]
-ldxdw r8, [r10-1848]
-jlt r8, r1, +0x2e8
+jne r2, 0, +0xa48
+mov64 r1, 12
+call mayhem_program::instructions::pump_interface::sell::<impl core::convert::From<mayhem_program::instructions::pump_interface::sell::SellError> for anchor_lang::error::Error>::from#5915
+ja +0x5a0
+jle r1, r9, +0x9e8
 mov64 r3, 0
-mov64 r9, 2
-jne r8, r1, +0x2f8
+jne r1, r9, +0x9f0
 and64 r2, 1
-jeq r2, 0, +0x4b8
-stxw [r10-312], r9
-mov64 r1, r10
-add64 r1, -280
-mov64 r6, r10
-add64 r6, -312
-mov64 r2, r6
-call mayhem_program::instructions::pump_amm_interface::pump_swap_types::PumpSwapError::name#fdf6
-stdw [r10-32], 0
-stdw [r10-40], 1
-stdw [r10-48], 0
-lddw r1, table[0:core::ptr::drop_in_place<alloc::string::String>#91b5, 3:<alloc::string::String as core::fmt::Write>::write_str#ca2d, 4:<alloc::string::String as core::fmt::Write>::write_char#6537, 5:core::fmt::Write::write_fmt#a290]
-stxdw [r10-120], r1
-mov64 r1, r10
-add64 r1, -48
-stxdw [r10-128], r1
-lddw r1, 0xe0000020
-stxdw [r10-112], r1
-mov64 r2, r10
-add64 r2, -128
+jeq r2, 0, +0xa08
+ldxdw r1, [r10-3336]
+and64 r1, 65535
+jgt r1, 10000, +0x2478
+ldxdw r1, [r8+48]
+ldxdw r6, [r1+104]
 mov64 r1, r6
-call <mayhem_program::instructions::pump_amm_interface::pump_swap_types::PumpSwapError as core::fmt::Display>::fmt#0036
-jne r0, 0, +0x47a0
-ldxdw r1, [r10-32]
-stxdw [r10-1760], r1
-ldxdw r1, [r10-40]
-stxdw [r10-1768], r1
-ldxdw r1, [r10-48]
-stxdw [r10-1776], r1
-ldxdw r1, [r10-280]
-stxdw [r10-1800], r1
-ldxdw r1, [r10-272]
-stxdw [r10-1792], r1
-ldxdw r1, [r10-264]
-stxdw [r10-1784], r1
-or64 r9, 6000
-stxw [r10-1680], r9
-stb [r10-1752], 2
-stdw [r10-1832], 2
-mov64 r1, r10
-add64 r1, -1832
-call <anchor_lang::error::Error as core::convert::From<anchor_lang::error::AnchorError>>::from#5ca3
-exit
-ldxdw r5, [r10-1840]
-mov64 r4, r5
-add64 r4, r0
-mov64 r3, 1
-mov64 r2, 1
-jlt r4, r5, +0x538
 mov64 r2, 0
-mov64 r4, r8
-add64 r4, r1
-add64 r4, r2
-jge r4, r8, +0x558
-jeq r4, r8, +0x568
-mov64 r2, r3
-and64 r2, 1
-jeq r2, 0, +0x578
-ja +0x358
-mov64 r4, r8
-add64 r4, r1
-add64 r4, r2
-jlt r4, r8, +0x510
-mov64 r3, 0
-jne r4, r8, +0x518
-and64 r2, 1
-jne r2, 0, +0x358
-mov64 r3, r0
-mov64 r2, 1
-mov64 r4, 1
-ldxdw r5, [r10-1840]
-jlt r5, r3, +0x5a8
-mov64 r4, 0
-mov64 r5, r8
-sub64 r5, r1
-sub64 r5, r4
-ldxdw r6, [r10-1840]
-sub64 r6, r0
-mov64 r4, 1
-ldxdw r9, [r10-1888]
-jlt r9, r6, +0x658
-mov64 r4, 0
-mov64 r6, 1
-ldxdw r9, [r10-1856]
-jge r9, r5, +0x670
-ldxdw r9, [r10-1856]
-jeq r9, r5, +0x688
-mov64 r4, r6
-ldxdw r5, [r10-1840]
-add64 r0, r5
-jge r0, r3, +0x6a0
-mov64 r9, 12
-and64 r4, 1
-jeq r4, 0, +0x6c0
-ja +0x358
-mov64 r6, 1
-ldxdw r9, [r10-1856]
-jlt r9, r5, +0x608
-mov64 r6, 0
-ldxdw r9, [r10-1856]
-jne r9, r5, +0x618
-ldxdw r5, [r10-1840]
-add64 r0, r5
-jlt r0, r3, +0x638
-mov64 r2, 0
-mov64 r9, 12
-and64 r4, 1
-jne r4, 0, +0x358
-add64 r1, r8
-add64 r1, r2
-mov64 r3, 1
-mov64 r2, 1
-ldxdw r4, [r10-1888]
-jle r4, r0, +0x738
-mov64 r2, 0
-ldxdw r4, [r10-1856]
-jgt r4, r1, +0x748
-ldxdw r4, [r10-1856]
-jeq r4, r1, +0x760
-mov64 r2, r3
-and64 r2, 1
-jne r2, 0, +0x770
-ja +0x358
-ldxdw r4, [r10-1856]
-jle r4, r1, +0x708
-mov64 r3, 0
-ldxdw r4, [r10-1856]
-jne r4, r1, +0x718
-and64 r2, 1
-jeq r2, 0, +0x358
-ldxdw r1, [r10-1896]
-and64 r1, 65535
-jge r1, 10001, +0x4890
-ldxdw r1, [r7+48]
-ldxdw r9, [r1+104]
-ldxdw r6, [r7+8]
-ldxdw r1, [r7+32]
-ldxdw r1, [r1]
-ldxdw r2, [r1+24]
-stxdw [r10-1856], r2
-ldxdw r8, [r1+16]
-ldxdw r2, [r1+8]
-stxdw [r10-1848], r2
-ldxdw r1, [r1]
-stxdw [r10-1840], r1
-ldxdw r7, [r10-1904]
-mov64 r1, r7
-mov64 r2, 0
-mov64 r3, r9
+ldxdw r3, [r10-3360]
 mov64 r4, 0
 call __multi3
 mov64 r2, r1
-mov64 r3, r9
-ldxdw r1, [r10-1872]
+mov64 r3, r6
+ldxdw r1, [r10-3352]
 add64 r3, r1
 mov64 r4, 1
-stxdw [r10-3552], r9
-jlt r3, r9, +0x858
+stxdw [r10-3320], r6
+jlt r3, r6, +0xad8
 mov64 r4, 0
 mov64 r1, r0
 call __udivti3
 mov64 r9, r0
-ldxdw r4, [r10-1896]
-jne r1, 0, +0x47f0
-add64 r6, 8
-mov64 r1, 96
-ldxdw r2, [r10-1840]
-jne r2, 0, +0x8c8
-ldxdw r2, [r10-1848]
-jne r2, 0, +0x8c8
-jne r8, 0, +0x8c8
-ldxdw r2, [r10-1856]
-jeq r2, 0, +0x948
-lddw r2, 0x8481abfe57889b06
-ldxdw r3, [r10-1840]
-jne r3, r2, +0x940
-lddw r2, 0x35c01846637f68fb
-ldxdw r3, [r10-1848]
-jne r3, r2, +0x940
-lddw r2, 0x553beb1adc39c4da
-jne r8, r2, +0x940
-lddw r2, 0x100000000f0a098
-ldxdw r3, [r10-1856]
-jeq r3, r2, +0x948
-mov64 r1, 176
-add64 r6, r1
-ldxdw r2, [r6]
-mov64 r3, r2
-jlt r2, r9, +0x970
-mov64 r3, r9
-ldxdw r1, [r10-1880]
+jne r1, 0, +0x23c0
+mov64 r1, 112
+ldxdw r2, [r10-3328]
+jne r2, 0, +0xb18
+mov64 r1, 168
+mov64 r3, 10000
+ldxdw r2, [r8+8]
+add64 r2, r1
+ldxdw r1, [r2]
+mov64 r2, r1
+jlt r1, r9, +0xb50
+mov64 r2, r9
+ldxdw r4, [r10-3336]
+sub64 r3, r4
+jeq r1, 0, +0xb70
+mov64 r9, r2
+ldxdw r1, [r10-3376]
 ldxdw r1, [r1+96]
-jeq r2, 0, +0x990
-mov64 r9, r3
-mov64 r3, 10000
-sub64 r3, r4
 and64 r3, 65535
 mov64 r2, 0
 mov64 r4, 0
@@ -308,1899 +361,678 @@
 mov64 r3, 10000
 mov64 r4, 0
 call __udivti3
-jlt r9, r0, +0xb40
+jlt r9, r0, +0xbd8
 mov64 r9, r0
 mov64 r6, r9
-add64 r6, r7
-mov64 r7, 1
-jge r6, r9, +0xb60
+ldxdw r1, [r10-3360]
+add64 r6, r1
+mov64 r8, 1
+jlt r6, r9, +0xc08
+mov64 r8, 0
 mov64 r1, r6
-or64 r1, r7
-jeq r1, 0, +0xb80
-stxdw [r10-2072], r9
+or64 r1, r8
+jeq r1, 0, +0x2410
 mov64 r1, r9
 mov64 r2, 0
-ldxdw r3, [r10-1872]
+ldxdw r3, [r10-3352]
 mov64 r4, 0
 call __multi3
 mov64 r2, r1
 mov64 r1, r0
 mov64 r3, r6
-mov64 r4, r7
+mov64 r4, r8
 call __udivti3
-jne r1, 0, +0x4840
-mov64 r9, 1
-lddw r2, 0x100000000f0a098
+jne r1, 0, +0x2428
+ldxdw r1, [r10-3376]
+ldxdw r1, [r1+88]
+ldxdw r8, [r10-3368]
+jlt r1, r0, +0xca0
+mov64 r1, r0
+stxdw [r10-3312], r1
+jeq r7, 0, +0xce8
+ldxdw r1, [r8+64]
+call solana_account_info::AccountInfo::lamports#aa02
+stxdw [r10-3352], r0
+ldxdw r1, [r8+48]
+ldxdw r1, [r1+104]
+stxdw [r10-3320], r1
+ja +0xd00
+ldxdw r1, [r8+56]
+ldxdw r1, [r1+104]
+stxdw [r10-3352], r1
+ldxdw r2, [r8+96]
+mov64 r6, r10
+add64 r6, -3304
+mov64 r1, r6
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r1, [r8+24]
+ldxdw r2, [r1+88]
+mov64 r1, r10
+add64 r1, -3256
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r8+32]
+mov64 r1, r10
+add64 r1, -3208
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r8+80]
+mov64 r1, r10
+add64 r1, -3160
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r8+88]
+mov64 r1, r10
+add64 r1, -3112
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r8+40]
+mov64 r1, r10
+add64 r1, -3064
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r8+120]
+mov64 r1, r10
+add64 r1, -3016
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r8+128]
+mov64 r1, r10
+add64 r1, -2968
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r8+240]
+mov64 r1, r10
+add64 r1, -2920
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r8+248]
+mov64 r1, r10
+add64 r1, -2872
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r1, [r8+104]
+ldxdw r2, [r1]
+mov64 r1, r10
+add64 r1, -2824
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r1, [r8+136]
+ldxdw r2, [r1+32]
+mov64 r1, r10
+add64 r1, -2776
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r8+112]
+mov64 r1, r10
+add64 r1, -2728
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r8+64]
+mov64 r1, r10
+add64 r1, -2680
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r1, [r8+48]
+ldxdw r2, [r1+32]
+mov64 r1, r10
+add64 r1, -2632
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r1, [r8+56]
+ldxdw r2, [r1+32]
+mov64 r1, r10
+add64 r1, -2584
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r8+144]
+mov64 r1, r10
+add64 r1, -2536
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r8+152]
+mov64 r1, r10
+add64 r1, -2488
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r8+160]
+mov64 r1, r10
+add64 r1, -2440
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r8+184]
+mov64 r1, r10
+add64 r1, -2392
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r8+192]
+mov64 r1, r10
+add64 r1, -2344
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r8+200]
+mov64 r1, r10
+add64 r1, -2296
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r8+208]
+mov64 r1, r10
+add64 r1, -2248
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r8+216]
+mov64 r1, r10
+add64 r1, -2200
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r8+72]
+mov64 r1, r10
+add64 r1, -2152
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r8+168]
+mov64 r1, r10
+add64 r1, -2104
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r8+176]
+mov64 r1, r10
+add64 r1, -2056
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r1, [r10-3344]
+ldxb r1, [r1+34]
+stxb [r10-2001], r1
+mov64 r1, r10
+add64 r1, -2001
+stxdw [r10-1984], r1
+lddw r1, .rodata[736f6c2d7661756c]
+stxdw [r10-2000], r1
+lddw r1, .rodata[0156e0f693665acf]
+ldxdw r2, [r1]
+stxdw [r10-616], r2
+ldxdw r2, [r1+8]
+stxdw [r10-608], r2
+ldxdw r2, [r1+16]
+stxdw [r10-600], r2
+ldxdw r1, [r1+24]
+stxdw [r10-592], r1
+stdw [r10-1976], 1
+stdw [r10-1992], 9
+mov64 r1, r10
+add64 r1, -2000
+stxdw [r10-1968], r1
+stdw [r10-1960], 2
+mov64 r1, r10
+add64 r1, -1928
+mov64 r2, r6
+mov64 r3, 1296
+call memcpy
+mov64 r1, r10
+add64 r1, -1968
+stxdw [r10-632], r1
+stdw [r10-624], 1
+stdw [r10-1936], 0
+stdw [r10-1944], 8
+stdw [r10-1952], 0
+mov64 r1, r10
+add64 r1, -1952
+ldxdw r2, [r10-3312]
+mov64 r3, r9
+mov64 r4, 0
+call pump_cpi::pump_v2::pump::cpi::buy_v2#6557
+mov64 r6, r0
+mov64 r2, r1
+mov64 r1, r6
+call core::ptr::drop_in_place<core::result::Result<(),anchor_lang::error::Error>>#64a0
+jne r6, 2, +0x1270
+ldxdw r1, [r8+48]
+call anchor_lang::accounts::interface_account::InterfaceAccount<T>::reload#d655
+jne r0, 2, +0x5a0
+ldxdw r1, [r8+104]
+call anchor_lang::accounts::account::Account<T>::reload#fa41
+jne r0, 2, +0x5a0
+jeq r7, 0, +0x1300
+ldxdw r1, [r8+64]
+call solana_account_info::AccountInfo::lamports#aa02
+ldxdw r6, [r10-3352]
+ldxdw r9, [r10-3320]
+ja +0x1338
+mov64 r1, r10
+add64 r1, -552
+lddw r2, .rodata[08000000]
+call mayhem_program::instructions::pump_interface::buy::BuyError::name#3012
+mov64 r1, r10
+add64 r1, -528
+mov64 r2, 8
+call <T as alloc::string::SpecToString>::spec_to_string#f398
+lddw r1, .rodata[70726f6772616d73]
+stxdw [r10-576], r1
+stb [r10-504], 2
+stw [r10-560], 105
+stdw [r10-568], 61
+stdw [r10-584], 0
+stw [r10-432], 6008
+ja +0x588
+ldxdw r1, [r8+56]
+call anchor_lang::accounts::interface_account::InterfaceAccount<T>::reload#d655
+ldxdw r6, [r10-3352]
+ldxdw r9, [r10-3320]
+jne r0, 2, +0x5a0
+ldxdw r1, [r8+56]
+ldxdw r0, [r1+104]
+jlt r6, r0, +0x2490
+ldxdw r1, [r8+48]
+ldxdw r1, [r1+104]
+stxdw [r10-3312], r1
+jlt r1, r9, +0x24a8
+stxdw [r10-3320], r0
+sub64 r6, r0
+ldxdw r1, [r8+8]
+ldxdw r2, [r10-3328]
+jeq r2, 0, +0x1410
+ldxdw r2, [r1+72]
+mov64 r3, r2
+add64 r3, r6
+mov64 r5, 1
+jlt r3, r2, +0x13b8
+mov64 r5, 0
+ldxdw r4, [r1+80]
+mov64 r2, r4
+add64 r2, r5
+mov64 r5, r4
+xor64 r5, r2
+xor64 r4, -1
+and64 r4, r5
+jslt r4, 0, +0x1738
+stxdw [r1+72], r3
+mov64 r3, 80
+ja +0x1490
+ldxdw r2, [r1+88]
+mov64 r3, r2
+add64 r3, r6
+mov64 r5, 1
+jlt r3, r2, +0x1440
+mov64 r5, 0
+ldxdw r4, [r1+96]
+mov64 r2, r4
+add64 r2, r5
+mov64 r5, r4
+xor64 r5, r2
+xor64 r4, -1
+and64 r4, r5
+jslt r4, 0, +0x1738
+stxdw [r1+88], r3
+mov64 r3, 96
+add64 r1, r3
+stxdw [r1], r2
+ldxdw r1, [r8+16]
+ldxdw r3, [r1+40]
+mov64 r2, r3
+add64 r2, r6
+mov64 r5, 1
+jlt r2, r3, +0x14d8
+mov64 r5, 0
+ldxdw r4, [r1+48]
+mov64 r3, r4
+add64 r3, r5
+mov64 r5, r4
+xor64 r5, r3
+xor64 r4, -1
+and64 r4, r5
+jslt r4, 0, +0x1738
+ldxdw r0, [r10-3312]
+sub64 r0, r9
+stxdw [r1+40], r2
+stxdw [r1+48], r3
+ldxdw r1, [r8+16]
+ldxdw r2, [r1+56]
+mov64 r5, 1
+jlt r2, r0, +0x1560
+mov64 r5, 0
+ldxdw r4, [r1+64]
+mov64 r3, r4
+sub64 r3, r5
+mov64 r5, r4
+xor64 r5, r3
+and64 r4, r5
+jsle r4, -1, +0x1738
+sub64 r2, r0
+stxdw [r1+56], r2
+stxdw [r1+64], r3
+mov64 r5, 72
+ldxdw r1, [r10-3328]
+jne r1, 0, +0x15d0
+mov64 r5, 88
+mov64 r1, 128
+ldxdw r2, [r10-3328]
+jne r2, 0, +0x15f0
+mov64 r1, 176
+stxdw [r10-3312], r0
+stxdw [r10-3352], r6
+mov64 r3, 136
+ldxdw r2, [r10-3328]
+jne r2, 0, +0x1620
+mov64 r3, 184
+ldxdw r2, [r8+8]
+mov64 r4, r2
+stxdw [r10-3328], r5
+add64 r4, r5
+ldxdw r9, [r4]
+ldxdw r5, [r4+8]
+mov64 r4, 1
+mov64 r6, 1
+jslt r5, 0, +0x1748
+mov64 r6, 0
+jne r9, 0, +0x1750
+jeq r5, 0, +0x1688
+mov64 r4, r6
+mov64 r5, r2
+add64 r5, r3
+ldxdw r6, [r5]
+and64 r4, 1
+jne r4, 0, +0x16c8
+add64 r2, r1
+ldxdw r8, [r2]
+jle r8, r9, +0x19b8
+ldxdw r8, [r10-3368]
+ldxdw r2, [r8+16]
+ldxdw r4, [r2+48]
 mov64 r1, 1
-ldxdw r3, [r10-1856]
-jeq r3, r2, +0xbd8
+mov64 r5, 1
+jslt r4, 0, +0x1768
+mov64 r5, 0
+ldxdw r3, [r2+40]
+jne r3, 0, +0x1778
+jeq r4, 0, +0x1788
+mov64 r1, r5
+and64 r1, 1
+jeq r1, 0, +0x1798
+ja +0x1878
+call mayhem_program::instructions::pump_interface::buy::math#52f7
+ja +0x5a0
+jeq r9, 0, +0x1678
+mov64 r4, 0
+jne r5, 0, +0x1680
+ja +0x1688
+ldxdw r3, [r2+40]
+jeq r3, 0, +0x1710
 mov64 r1, 0
-lddw r3, 0x553beb1adc39c4da
-mov64 r2, 1
-jne r8, r3, +0xbf8
-lddw r4, 0x35c01846637f68fb
-mov64 r3, 1
-ldxdw r5, [r10-1848]
-jeq r5, r4, +0xc28
-mov64 r3, 0
-lddw r4, 0x8481abfe57889b06
-ldxdw r5, [r10-1840]
-jne r5, r4, +0xc48
-ja +0xc50
-mov64 r6, r9
-add64 r6, r7
-mov64 r7, 1
-jlt r6, r9, +0xa18
-mov64 r7, 0
-mov64 r1, r6
-or64 r1, r7
-jne r1, 0, +0xa30
-stw [r10-1832], 9
-mov64 r3, r10
-add64 r3, -1832
-lddw r1, str("called `Result::unwrap()` on an `Err` value")
-mov64 r2, 43
-lddw r4, table[3:<mayhem_program::instructions::pump_amm_interface::pump_swap_types::PumpSwapError as core::fmt::Debug>::fmt#b375]
-lddw r5, panic_location(programs/mayhem-program/src/state.rs:378:89)
-call core::result::unwrap_failed#5a05
-lddw r3, 0x553beb1adc39c4da
-mov64 r2, 1
-jeq r8, r3, +0xae8
-mov64 r2, 0
-lddw r4, 0x35c01846637f68fb
-mov64 r3, 1
-ldxdw r5, [r10-1848]
-jne r5, r4, +0xb10
-lddw r4, 0x8481abfe57889b06
-ldxdw r5, [r10-1840]
-jeq r5, r4, +0xc50
-mov64 r9, 0
-and64 r9, r3
-and64 r9, r2
-and64 r9, r1
-ldxdw r1, [r10-1880]
-ldxdw r7, [r1+88]
-ldxdw r6, [r10-1864]
-jlt r7, r0, +0xc90
-mov64 r7, r0
-and64 r9, 1
-jeq r9, 0, +0xcd8
-ldxdw r1, [r6+64]
-call solana_account_info::AccountInfo::lamports#aa02
-stxdw [r10-3560], r0
-ldxdw r1, [r6+48]
-ldxdw r1, [r1+104]
-stxdw [r10-3552], r1
-ja +0xcf0
-ldxdw r1, [r6+56]
-ldxdw r1, [r1+104]
-stxdw [r10-3560], r1
-ldxdw r1, [r6+176]
-ldxdw r1, [r1]
-ldxdw r2, [r1+24]
-stxdw [r10-24], r2
-ldxdw r2, [r1+16]
-stxdw [r10-32], r2
-ldxdw r2, [r1+8]
-stxdw [r10-40], r2
-ldxdw r1, [r1]
-stxdw [r10-48], r1
-ldxdw r2, [r6+96]
-ldxdw r0, [r2+8]
-ldxdw r1, [r0]
-add64 r1, 1
-mov64 r3, 1
-jeq r1, 0, +0xd78
-mov64 r3, 0
-ldxdw r4, [r2]
-stxdw [r0], r1
-jeq r3, 1, +0x4728
-ldxdw r5, [r2+16]
-ldxdw r1, [r5]
-add64 r1, 1
-mov64 r3, 1
-jeq r1, 0, +0xdc0
-mov64 r3, 0
-stxdw [r5], r1
-jeq r3, 1, +0x4728
-stxdw [r10-1968], r7
-stxdw [r10-2064], r4
-ldxdw r1, [r6+24]
-ldxdw r1, [r1+88]
-ldxdw r3, [r1+8]
-stxdw [r10-1840], r3
+jne r4, 0, +0x1718
+and64 r1, 1
+jne r1, 0, +0x1878
+jgt r6, r3, +0x1878
+mov64 r1, r10
+add64 r1, -552
+lddw r2, .rodata[07000000]
+call mayhem_program::instructions::pump_interface::buy::BuyError::name#3012
+mov64 r1, r10
+add64 r1, -528
+mov64 r2, 7
+call <T as alloc::string::SpecToString>::spec_to_string#f398
+lddw r1, .rodata[70726f6772616d73]
+stxdw [r10-576], r1
+stb [r10-504], 2
+stw [r10-560], 116
+stdw [r10-568], 61
+stdw [r10-584], 0
+stw [r10-432], 6007
+mov64 r1, r10
+add64 r1, -584
+call <anchor_lang::error::Error as core::convert::From<anchor_lang::error::AnchorError>>::from#5ca3
+mov64 r2, r1
+ldxdw r1, [r8+16]
+ldxdw r4, [r1+40]
+mov64 r1, r0
+mov64 r3, r6
+call anchor_lang::error::Error::with_values#4132
+ja +0x5a0
+ldxdw r1, [r8+64]
+ldxdw r3, [r1]
+ldxdw r4, [r3+24]
+stxdw [r10-264], r4
+ldxdw r4, [r3+16]
+stxdw [r10-272], r4
+ldxdw r4, [r3+8]
+stxdw [r10-280], r4
 ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0xe28
-mov64 r4, 0
-ldxb r7, [r2+42]
-stxdw [r10-2120], r7
-ldxb r7, [r2+41]
-stxdw [r10-2112], r7
-ldxb r7, [r2+40]
-stxdw [r10-2104], r7
-ldxdw r7, [r2+32]
-stxdw [r10-2096], r7
-ldxdw r2, [r2+24]
-stxdw [r10-2088], r2
-ldxdw r2, [r1]
-stxdw [r10-2080], r2
-ldxdw r2, [r10-1840]
-stxdw [r2], r3
-jeq r4, 1, +0x4728
-ldxdw r4, [r1+16]
-ldxdw r2, [r4]
-add64 r2, 1
-mov64 r3, 1
-jeq r2, 0, +0xed0
-mov64 r3, 0
-stxdw [r10-2128], r4
-stxdw [r4], r2
-jeq r3, 1, +0x4728
-ldxdw r2, [r6+32]
-ldxdw r3, [r2+8]
-stxdw [r10-1848], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0xf28
-mov64 r4, 0
-ldxb r7, [r1+42]
-stxdw [r10-2168], r7
-ldxb r7, [r1+41]
-stxdw [r10-2160], r7
-ldxb r7, [r1+40]
-stxdw [r10-2152], r7
-ldxdw r7, [r1+32]
-stxdw [r10-2144], r7
-ldxdw r1, [r1+24]
-stxdw [r10-2136], r1
-ldxdw r7, [r2]
-ldxdw r1, [r10-1848]
-stxdw [r1], r3
-jeq r4, 1, +0x4728
-ldxdw r4, [r2+16]
-ldxdw r1, [r4]
-add64 r1, 1
-mov64 r3, 1
-jeq r1, 0, +0xfc8
-mov64 r3, 0
-stxdw [r10-2176], r4
-stxdw [r4], r1
-jeq r3, 1, +0x4728
-stxdw [r10-2184], r7
-ldxdw r1, [r6+80]
-ldxdw r3, [r1+8]
-stxdw [r10-1856], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x1028
-mov64 r4, 0
-ldxb r7, [r2+42]
-stxdw [r10-2232], r7
-ldxb r7, [r2+41]
-stxdw [r10-2224], r7
-ldxb r7, [r2+40]
-stxdw [r10-2216], r7
-ldxdw r7, [r2+32]
-stxdw [r10-2208], r7
-ldxdw r2, [r2+24]
-stxdw [r10-2200], r2
-ldxdw r2, [r1]
-stxdw [r10-2192], r2
-ldxdw r2, [r10-1856]
-stxdw [r2], r3
-jeq r4, 1, +0x4728
-ldxdw r4, [r1+16]
-ldxdw r2, [r4]
-add64 r2, 1
-mov64 r3, 1
-jeq r2, 0, +0x10d0
-mov64 r3, 0
-stxdw [r10-2240], r4
-stxdw [r4], r2
-jeq r3, 1, +0x4728
-ldxdw r2, [r6+88]
-ldxdw r3, [r2+8]
-stxdw [r10-1872], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x1128
-mov64 r4, 0
-ldxb r7, [r1+42]
-stxdw [r10-2280], r7
-ldxb r7, [r1+41]
-stxdw [r10-2272], r7
-ldxb r7, [r1+40]
-stxdw [r10-2264], r7
-ldxdw r7, [r1+32]
-stxdw [r10-2256], r7
-ldxdw r1, [r1+24]
-stxdw [r10-2248], r1
-ldxdw r7, [r2]
-ldxdw r1, [r10-1872]
-stxdw [r1], r3
-jeq r4, 1, +0x4728
-ldxdw r4, [r2+16]
-ldxdw r1, [r4]
-add64 r1, 1
-mov64 r3, 1
-jeq r1, 0, +0x11c8
-mov64 r3, 0
-stxdw [r10-2288], r4
-stxdw [r4], r1
-jeq r3, 1, +0x4728
-stxdw [r10-2296], r7
-ldxdw r1, [r6+40]
-ldxdw r3, [r1+8]
-stxdw [r10-1880], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x1228
-mov64 r4, 0
-ldxb r7, [r2+42]
-stxdw [r10-2344], r7
-ldxb r7, [r2+41]
-stxdw [r10-2336], r7
-ldxb r7, [r2+40]
-stxdw [r10-2328], r7
-ldxdw r7, [r2+32]
-stxdw [r10-2320], r7
-ldxdw r2, [r2+24]
-stxdw [r10-2312], r2
-ldxdw r2, [r1]
-stxdw [r10-2304], r2
-ldxdw r2, [r10-1880]
-stxdw [r2], r3
-jeq r4, 1, +0x4728
-ldxdw r4, [r1+16]
-ldxdw r2, [r4]
-add64 r2, 1
-mov64 r3, 1
-jeq r2, 0, +0x12d0
-mov64 r3, 0
-stxdw [r10-2352], r4
-stxdw [r4], r2
-jeq r3, 1, +0x4728
-ldxdw r2, [r6+120]
-ldxdw r3, [r2+8]
-stxdw [r10-1888], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x1328
-mov64 r4, 0
-ldxb r7, [r1+42]
-stxdw [r10-2392], r7
-ldxb r7, [r1+41]
-stxdw [r10-2384], r7
-ldxb r7, [r1+40]
-stxdw [r10-2376], r7
-ldxdw r7, [r1+32]
-stxdw [r10-2368], r7
-ldxdw r1, [r1+24]
-stxdw [r10-2360], r1
-ldxdw r7, [r2]
-ldxdw r1, [r10-1888]
-stxdw [r1], r3
-jeq r4, 1, +0x4728
-ldxdw r4, [r2+16]
-ldxdw r1, [r4]
-add64 r1, 1
-mov64 r3, 1
-jeq r1, 0, +0x13c8
-mov64 r3, 0
-stxdw [r10-2400], r4
-stxdw [r4], r1
-jeq r3, 1, +0x4728
-stxdw [r10-2408], r7
-ldxdw r1, [r6+128]
-ldxdw r3, [r1+8]
-stxdw [r10-1904], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x1428
-mov64 r4, 0
-ldxb r7, [r2+42]
-stxdw [r10-2456], r7
-ldxb r7, [r2+41]
-stxdw [r10-2448], r7
-ldxb r7, [r2+40]
-stxdw [r10-2440], r7
-ldxdw r7, [r2+32]
-stxdw [r10-2432], r7
-ldxdw r2, [r2+24]
-stxdw [r10-2424], r2
-ldxdw r2, [r1]
-stxdw [r10-2416], r2
-ldxdw r2, [r10-1904]
-stxdw [r2], r3
-jeq r4, 1, +0x4728
-ldxdw r4, [r1+16]
-ldxdw r2, [r4]
-add64 r2, 1
-mov64 r3, 1
-jeq r2, 0, +0x14d0
-mov64 r3, 0
-stxdw [r10-2464], r4
-stxdw [r4], r2
-jeq r3, 1, +0x4728
-ldxdw r2, [r6+240]
-ldxdw r3, [r2+8]
-stxdw [r10-1920], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x1528
-mov64 r4, 0
-ldxb r7, [r1+42]
-stxdw [r10-2504], r7
-ldxb r7, [r1+41]
-stxdw [r10-2496], r7
-ldxb r7, [r1+40]
-stxdw [r10-2488], r7
-ldxdw r7, [r1+32]
-stxdw [r10-2480], r7
-ldxdw r1, [r1+24]
-stxdw [r10-2472], r1
-ldxdw r8, [r2]
-ldxdw r1, [r10-1920]
-stxdw [r1], r3
-jeq r4, 1, +0x4728
-ldxdw r4, [r2+16]
-ldxdw r1, [r4]
-add64 r1, 1
-mov64 r3, 1
-jeq r1, 0, +0x15c8
-mov64 r3, 0
-stxdw [r10-2512], r4
-stxdw [r4], r1
-jeq r3, 1, +0x4728
-ldxdw r1, [r6+248]
-ldxdw r3, [r1+8]
-stxdw [r10-1928], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x1620
-mov64 r4, 0
-ldxb r7, [r2+42]
-stxdw [r10-2560], r7
-ldxb r7, [r2+41]
-stxdw [r10-2552], r7
-ldxb r7, [r2+40]
-stxdw [r10-2544], r7
-ldxdw r7, [r2+32]
-stxdw [r10-2536], r7
-ldxdw r2, [r2+24]
-stxdw [r10-2528], r2
-ldxdw r2, [r1]
-stxdw [r10-2520], r2
-ldxdw r2, [r10-1928]
-stxdw [r2], r3
-jeq r4, 1, +0x4728
-ldxdw r4, [r1+16]
-ldxdw r2, [r4]
-add64 r2, 1
-mov64 r3, 1
-jeq r2, 0, +0x16c8
-mov64 r3, 0
-stxdw [r10-2568], r4
-stxdw [r4], r2
-jeq r3, 1, +0x4728
-ldxdw r2, [r6+104]
-ldxdw r2, [r2]
-ldxdw r3, [r2+8]
-stxdw [r10-1936], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x1728
-mov64 r4, 0
-ldxb r7, [r1+42]
-stxdw [r10-2608], r7
-ldxb r7, [r1+41]
-stxdw [r10-2600], r7
-ldxb r7, [r1+40]
-stxdw [r10-2592], r7
-ldxdw r7, [r1+32]
-stxdw [r10-2584], r7
-ldxdw r1, [r1+24]
-stxdw [r10-2576], r1
-ldxdw r7, [r2]
-ldxdw r1, [r10-1936]
-stxdw [r1], r3
-jeq r4, 1, +0x4728
-ldxdw r4, [r2+16]
-ldxdw r1, [r4]
-add64 r1, 1
-mov64 r3, 1
-jeq r1, 0, +0x17c8
-mov64 r3, 0
-stxdw [r10-2616], r4
-stxdw [r4], r1
-jeq r3, 1, +0x4728
-stxdw [r10-2624], r8
-ldxdw r1, [r6+136]
-ldxdw r1, [r1+32]
-ldxdw r3, [r1+8]
-stxdw [r10-1944], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x1830
-mov64 r4, 0
-ldxb r8, [r2+42]
-stxdw [r10-2672], r8
-ldxb r8, [r2+41]
-stxdw [r10-2664], r8
-ldxb r8, [r2+40]
-stxdw [r10-2656], r8
-ldxdw r8, [r2+32]
-stxdw [r10-2648], r8
-ldxdw r2, [r2+24]
-stxdw [r10-2640], r2
-ldxdw r2, [r1]
-stxdw [r10-2632], r2
-ldxdw r2, [r10-1944]
-stxdw [r2], r3
-jeq r4, 1, +0x4728
-ldxdw r4, [r1+16]
-ldxdw r2, [r4]
-add64 r2, 1
-mov64 r3, 1
-jeq r2, 0, +0x18d8
-mov64 r3, 0
-stxdw [r10-2680], r4
-stxdw [r4], r2
-jeq r3, 1, +0x4728
-ldxdw r2, [r6+112]
-ldxdw r3, [r2+8]
-stxdw [r10-1952], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x1930
-mov64 r4, 0
-ldxb r8, [r1+42]
-stxdw [r10-2728], r8
-ldxb r8, [r1+41]
-stxdw [r10-2720], r8
-ldxb r8, [r1+40]
-stxdw [r10-2712], r8
-ldxdw r8, [r1+32]
-stxdw [r10-2704], r8
-ldxdw r1, [r1+24]
-stxdw [r10-2696], r1
-ldxdw r1, [r2]
-stxdw [r10-2688], r1
-ldxdw r1, [r10-1952]
-stxdw [r1], r3
-jeq r4, 1, +0x4728
-ldxdw r4, [r2+16]
-ldxdw r1, [r4]
-add64 r1, 1
-mov64 r3, 1
-jeq r1, 0, +0x19d8
-mov64 r3, 0
-stxdw [r10-2736], r4
-stxdw [r4], r1
-jeq r3, 1, +0x4728
-ldxdw r1, [r6+64]
-ldxdw r3, [r1+8]
-stxdw [r10-1960], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x1a30
-mov64 r4, 0
-ldxb r8, [r2+42]
-stxdw [r10-2784], r8
-ldxb r8, [r2+41]
-stxdw [r10-2776], r8
-ldxb r8, [r2+40]
-stxdw [r10-2768], r8
-ldxdw r8, [r2+32]
-stxdw [r10-2760], r8
-ldxdw r2, [r2+24]
-stxdw [r10-2752], r2
-ldxdw r2, [r1]
-stxdw [r10-2744], r2
-ldxdw r2, [r10-1960]
-stxdw [r2], r3
-jeq r4, 1, +0x4728
-ldxdw r4, [r1+16]
-ldxdw r2, [r4]
-add64 r2, 1
-mov64 r3, 1
-jeq r2, 0, +0x1ad8
-mov64 r3, 0
-stxdw [r10-2792], r4
-stxdw [r4], r2
-jeq r3, 1, +0x4728
-ldxdw r2, [r6+48]
-ldxdw r2, [r2+32]
-ldxdw r3, [r2+8]
-stxdw [r10-1976], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x1b38
-mov64 r4, 0
-ldxb r8, [r1+42]
-stxdw [r10-2832], r8
-ldxb r8, [r1+41]
-stxdw [r10-2824], r8
-ldxb r8, [r1+40]
-stxdw [r10-2816], r8
-ldxdw r8, [r1+32]
-stxdw [r10-2808], r8
-ldxdw r1, [r1+24]
-stxdw [r10-2800], r1
-ldxdw r8, [r2]
-ldxdw r1, [r10-1976]
-stxdw [r1], r3
-jeq r4, 1, +0x4728
-ldxdw r4, [r2+16]
-ldxdw r1, [r4]
-add64 r1, 1
-mov64 r3, 1
-jeq r1, 0, +0x1bd8
-mov64 r3, 0
-stxdw [r10-2840], r4
-stxdw [r4], r1
-jeq r3, 1, +0x4728
-stxdw [r10-2848], r7
-ldxdw r1, [r6+56]
-ldxdw r1, [r1+32]
-ldxdw r3, [r1+8]
-stxdw [r10-1984], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x1c40
-mov64 r4, 0
-ldxb r7, [r2+42]
-stxdw [r10-2896], r7
-ldxb r7, [r2+41]
-stxdw [r10-2888], r7
-ldxb r7, [r2+40]
-stxdw [r10-2880], r7
-ldxdw r7, [r2+32]
-stxdw [r10-2872], r7
-ldxdw r2, [r2+24]
-stxdw [r10-2864], r2
-ldxdw r2, [r1]
-stxdw [r10-2856], r2
-ldxdw r2, [r10-1984]
-stxdw [r2], r3
-jeq r4, 1, +0x4728
-ldxdw r4, [r1+16]
-ldxdw r2, [r4]
-add64 r2, 1
-mov64 r3, 1
-jeq r2, 0, +0x1ce8
-mov64 r3, 0
-stxdw [r10-2904], r4
-stxdw [r4], r2
-jeq r3, 1, +0x4728
-ldxdw r2, [r6+144]
-ldxdw r3, [r2+8]
-stxdw [r10-1992], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x1d40
-mov64 r4, 0
-ldxb r7, [r1+42]
-stxdw [r10-2944], r7
-ldxb r7, [r1+41]
-stxdw [r10-2936], r7
-ldxb r7, [r1+40]
-stxdw [r10-2928], r7
-ldxdw r7, [r1+32]
-stxdw [r10-2920], r7
-ldxdw r1, [r1+24]
-stxdw [r10-2912], r1
-ldxdw r7, [r2]
-ldxdw r1, [r10-1992]
-stxdw [r1], r3
-jeq r4, 1, +0x4728
-ldxdw r4, [r2+16]
-ldxdw r1, [r4]
-add64 r1, 1
-mov64 r3, 1
-jeq r1, 0, +0x1de0
-mov64 r3, 0
-stxdw [r10-2952], r4
-stxdw [r4], r1
-jeq r3, 1, +0x4728
-stxdw [r10-2960], r8
-ldxdw r1, [r6+152]
-ldxdw r3, [r1+8]
-stxdw [r10-2000], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x1e40
-mov64 r4, 0
-ldxb r8, [r2+42]
-stxdw [r10-3008], r8
-ldxb r8, [r2+41]
-stxdw [r10-3000], r8
-ldxb r8, [r2+40]
-stxdw [r10-2992], r8
-ldxdw r8, [r2+32]
-stxdw [r10-2984], r8
-ldxdw r2, [r2+24]
-stxdw [r10-2976], r2
-ldxdw r2, [r1]
-stxdw [r10-2968], r2
-ldxdw r2, [r10-2000]
-stxdw [r2], r3
-jeq r4, 1, +0x4728
-ldxdw r4, [r1+16]
-ldxdw r2, [r4]
-add64 r2, 1
-mov64 r3, 1
-jeq r2, 0, +0x1ee8
-mov64 r3, 0
-stxdw [r10-3016], r4
-stxdw [r4], r2
-jeq r3, 1, +0x4728
-stxdw [r10-3024], r7
-ldxdw r2, [r6+160]
-ldxdw r3, [r2+8]
-stxdw [r10-2008], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x1f48
-mov64 r4, 0
-ldxb r8, [r1+42]
-stxdw [r10-3064], r8
-ldxb r8, [r1+41]
-stxdw [r10-3056], r8
-ldxb r8, [r1+40]
-stxdw [r10-3048], r8
-ldxdw r8, [r1+32]
-stxdw [r10-3040], r8
-ldxdw r1, [r1+24]
-stxdw [r10-3032], r1
-ldxdw r7, [r2]
-ldxdw r1, [r10-2008]
-stxdw [r1], r3
-jeq r4, 1, +0x4728
-ldxdw r4, [r2+16]
-ldxdw r1, [r4]
-add64 r1, 1
-mov64 r3, 1
-jeq r1, 0, +0x1fe8
-mov64 r3, 0
-mov64 r8, r5
-stxdw [r10-3072], r4
-stxdw [r4], r1
-jeq r3, 1, +0x4728
-mov64 r5, r0
-ldxdw r1, [r6+184]
-ldxdw r3, [r1+8]
-stxdw [r10-2016], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x2050
-mov64 r4, 0
-ldxb r0, [r2+42]
-stxdw [r10-3120], r0
-ldxb r0, [r2+41]
-stxdw [r10-3112], r0
-ldxb r0, [r2+40]
-stxdw [r10-3104], r0
-ldxdw r0, [r2+32]
-stxdw [r10-3096], r0
-ldxdw r2, [r2+24]
-stxdw [r10-3088], r2
-ldxdw r2, [r1]
-stxdw [r10-3080], r2
-ldxdw r2, [r10-2016]
-stxdw [r2], r3
-jeq r4, 1, +0x4728
-ldxdw r4, [r1+16]
-ldxdw r2, [r4]
-add64 r2, 1
-mov64 r3, 1
-mov64 r0, r5
-jeq r2, 0, +0x2100
-mov64 r3, 0
-stxdw [r10-3128], r4
-stxdw [r4], r2
-mov64 r5, r8
-jeq r3, 1, +0x4728
-ldxdw r2, [r6+192]
-ldxdw r3, [r2+8]
-stxdw [r10-2024], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x2160
-mov64 r4, 0
-ldxb r8, [r1+42]
-stxdw [r10-3168], r8
-ldxb r8, [r1+41]
-stxdw [r10-3160], r8
-ldxb r8, [r1+40]
-stxdw [r10-3152], r8
-ldxdw r8, [r1+32]
-stxdw [r10-3144], r8
-ldxdw r1, [r1+24]
-stxdw [r10-3136], r1
-ldxdw r8, [r2]
-ldxdw r1, [r10-2024]
-stxdw [r1], r3
-jeq r4, 1, +0x4728
-ldxdw r4, [r2+16]
-ldxdw r1, [r4]
-add64 r1, 1
-mov64 r3, 1
-jeq r1, 0, +0x2200
-mov64 r3, 0
-stxdw [r10-3176], r4
-stxdw [r4], r1
-jeq r3, 1, +0x4728
-stxdw [r10-3184], r7
-ldxdw r1, [r6+200]
-ldxdw r3, [r1+8]
-stxdw [r10-2032], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x2260
-mov64 r4, 0
-ldxb r7, [r2+42]
-stxdw [r10-3232], r7
-ldxb r7, [r2+41]
-stxdw [r10-3224], r7
-ldxb r7, [r2+40]
-stxdw [r10-3216], r7
-ldxdw r7, [r2+32]
-stxdw [r10-3208], r7
-ldxdw r2, [r2+24]
-stxdw [r10-3200], r2
-ldxdw r2, [r1]
-stxdw [r10-3192], r2
-ldxdw r2, [r10-2032]
-stxdw [r2], r3
-jeq r4, 1, +0x4728
-ldxdw r4, [r1+16]
-ldxdw r2, [r4]
-add64 r2, 1
-mov64 r3, 1
-jeq r2, 0, +0x2308
-mov64 r3, 0
-stxdw [r10-3240], r4
-stxdw [r4], r2
-jeq r3, 1, +0x4728
-ldxdw r3, [r6+208]
-ldxdw r2, [r3+8]
-stxdw [r10-2040], r2
-ldxdw r2, [r2]
-add64 r2, 1
-mov64 r4, 1
-jeq r2, 0, +0x2360
-mov64 r4, 0
-ldxb r7, [r1+42]
-stxdw [r10-3280], r7
-ldxb r7, [r1+41]
-stxdw [r10-3272], r7
-ldxb r7, [r1+40]
-stxdw [r10-3264], r7
-ldxdw r7, [r1+32]
-stxdw [r10-3256], r7
-ldxdw r1, [r1+24]
-stxdw [r10-3248], r1
-ldxdw r7, [r3]
-ldxdw r1, [r10-2040]
-stxdw [r1], r2
-jeq r4, 1, +0x4728
-ldxdw r4, [r3+16]
-ldxdw r1, [r4]
-add64 r1, 1
-mov64 r2, 1
-jeq r1, 0, +0x2400
-mov64 r2, 0
-stxdw [r10-3288], r5
-stxdw [r10-3296], r4
-stxdw [r4], r1
-jeq r2, 1, +0x4728
-stxdw [r10-3304], r8
-ldxdw r5, [r6+216]
-ldxdw r1, [r5+8]
-ldxdw r2, [r1]
-add64 r2, 1
-mov64 r4, 1
-jeq r2, 0, +0x2460
-mov64 r4, 0
-ldxb r8, [r3+42]
-stxdw [r10-3352], r8
-ldxb r8, [r3+41]
-stxdw [r10-3344], r8
-ldxb r8, [r3+40]
-stxdw [r10-3336], r8
-ldxdw r8, [r3+32]
-stxdw [r10-3328], r8
-ldxdw r3, [r3+24]
-stxdw [r10-3320], r3
-ldxdw r3, [r5]
-stxdw [r10-3312], r1
-stxdw [r1], r2
-jeq r4, 1, +0x4728
-stxdw [r10-3368], r3
-ldxdw r1, [r5+16]
-stxdw [r10-2048], r1
-ldxdw r1, [r1]
-add64 r1, 1
-mov64 r2, 1
-jeq r1, 0, +0x2510
-mov64 r2, 0
-stxdw [r10-3360], r7
-ldxdw r4, [r10-2064]
-mov64 r3, r0
-ldxdw r0, [r10-2048]
-stxdw [r0], r1
-jeq r2, 1, +0x4728
-ldxdw r7, [r6+72]
-ldxdw r1, [r7+8]
-ldxdw r2, [r1]
-add64 r2, 1
-mov64 r0, 1
-jeq r2, 0, +0x2578
-mov64 r0, 0
-ldxb r8, [r5+42]
-stxdw [r10-3408], r8
-ldxb r8, [r5+41]
-stxdw [r10-3400], r8
-ldxb r8, [r5+40]
-stxdw [r10-3392], r8
-ldxdw r8, [r5+32]
-stxdw [r10-3384], r8
-ldxdw r5, [r5+24]
-ldxdw r8, [r7]
-stxdw [r10-3376], r1
-stxdw [r1], r2
-jeq r0, 1, +0x4728
-ldxdw r0, [r7+16]
-ldxdw r1, [r0]
-add64 r1, 1
-mov64 r2, 1
-jeq r1, 0, +0x2610
-mov64 r2, 0
-stxdw [r10-3416], r5
-stxdw [r10-3568], r9
-stxdw [r0], r1
-jeq r2, 1, +0x4728
-ldxdw r9, [r6+168]
-ldxdw r6, [r9+8]
-ldxdw r2, [r6]
-add64 r2, 1
-mov64 r5, 1
-jeq r2, 0, +0x2668
-mov64 r5, 0
-ldxb r1, [r7+42]
-stxdw [r10-3464], r1
-ldxb r1, [r7+41]
-stxdw [r10-3456], r1
-ldxb r1, [r7+40]
-stxdw [r10-3448], r1
-ldxdw r1, [r7+32]
-stxdw [r10-3440], r1
-ldxdw r1, [r7+24]
-stxdw [r10-3432], r1
-ldxdw r1, [r9]
-stxdw [r10-3424], r1
-stxdw [r6], r2
-jeq r5, 1, +0x4728
-ldxdw r1, [r9+16]
-stxdw [r10-2056], r1
-ldxdw r1, [r1]
-add64 r1, 1
-mov64 r2, 1
-ldxdw r7, [r10-1968]
-jeq r1, 0, +0x2718
-mov64 r2, 0
-mov64 r5, r6
-stxdw [r10-3472], r8
-ldxdw r6, [r10-2056]
-stxdw [r6], r1
-jeq r2, 1, +0x4728
-mov64 r6, r0
-ldxdw r1, [r10-1864]
-ldxdw r2, [r1+176]
-ldxdw r7, [r2+8]
-ldxdw r8, [r7]
-add64 r8, 1
-mov64 r1, 1
-jeq r8, 0, +0x2788
-mov64 r1, 0
-ldxb r0, [r9+42]
-stxdw [r10-3520], r0
-ldxb r0, [r9+41]
-stxdw [r10-3512], r0
-ldxb r0, [r9+40]
-stxdw [r10-3504], r0
-ldxdw r0, [r9+32]
-stxdw [r10-3496], r0
-ldxdw r9, [r9+24]
-stxdw [r10-3488], r9
-ldxdw r9, [r2]
-stxdw [r10-3480], r9
-stxdw [r7], r8
-jeq r1, 1, +0x4728
-ldxdw r9, [r2+16]
-ldxdw r1, [r9]
-add64 r1, 1
-mov64 r8, 1
-mov64 r0, r6
-jeq r1, 0, +0x2830
-mov64 r8, 0
-stxdw [r9], r1
-mov64 r6, r5
-jeq r8, 1, +0x4728
-ldxdw r5, [r2+24]
-ldxdw r8, [r2+32]
-ldxb r1, [r2+40]
-stxdw [r10-3528], r1
-ldxb r1, [r2+41]
-stxdw [r10-3536], r1
-ldxb r1, [r2+42]
-stxdw [r10-3544], r1
-ldxdw r1, [r10-1912]
-ldxb r1, [r1+34]
-mov64 r2, r10
-add64 r2, -313
-stxdw [r10-112], r2
-lddw r2, .rodata[736f6c2d7661756c]
-stxdw [r10-128], r2
-stxb [r10-313], r1
-mov64 r1, r10
-add64 r1, -128
-stxdw [r10-336], r1
-ldxdw r1, [r10-2120]
-stxb [r10-1766], r1
-ldxdw r1, [r10-2112]
-stxb [r10-1767], r1
-ldxdw r1, [r10-2104]
-stxb [r10-1768], r1
-ldxdw r1, [r10-2096]
-stxdw [r10-1776], r1
-ldxdw r1, [r10-2088]
-stxdw [r10-1784], r1
-ldxdw r1, [r10-3288]
-stxdw [r10-1792], r1
-stxdw [r10-1800], r3
-stxdw [r10-1808], r4
-stdw [r10-104], 1
-stdw [r10-120], 9
-stdw [r10-328], 2
-ldxw r1, [r10-280]
-stxw [r10-1765], r1
-ldxb r1, [r10-276]
-stxb [r10-1761], r1
-ldxdw r1, [r10-2168]
-stxb [r10-1718], r1
-ldxdw r1, [r10-2160]
-stxb [r10-1719], r1
-ldxdw r1, [r10-2152]
-stxb [r10-1720], r1
-ldxdw r1, [r10-2144]
-stxdw [r10-1728], r1
-ldxdw r1, [r10-2136]
-stxdw [r10-1736], r1
-ldxdw r1, [r10-2128]
-stxdw [r10-1744], r1
-ldxdw r1, [r10-1840]
-stxdw [r10-1752], r1
-ldxdw r1, [r10-2080]
-stxdw [r10-1760], r1
-ldxb r1, [r10-308]
-stxb [r10-1713], r1
-ldxw r1, [r10-312]
-stxw [r10-1717], r1
-ldxdw r1, [r10-2232]
-stxb [r10-1670], r1
-ldxdw r1, [r10-2224]
-stxb [r10-1671], r1
-ldxdw r1, [r10-2216]
-stxb [r10-1672], r1
-ldxdw r1, [r10-2208]
-stxdw [r10-1680], r1
-ldxdw r1, [r10-2200]
-stxdw [r10-1688], r1
-ldxdw r1, [r10-2176]
-stxdw [r10-1696], r1
-ldxdw r1, [r10-1848]
-stxdw [r10-1704], r1
-ldxdw r1, [r10-2184]
-stxdw [r10-1712], r1
-ldxb r1, [r10-337]
-stxb [r10-1665], r1
-ldxw r1, [r10-341]
-stxw [r10-1669], r1
-ldxdw r1, [r10-2280]
-stxb [r10-1622], r1
-ldxdw r1, [r10-2272]
-stxb [r10-1623], r1
-ldxdw r1, [r10-2264]
-stxb [r10-1624], r1
-ldxdw r1, [r10-2256]
-stxdw [r10-1632], r1
-ldxdw r1, [r10-2248]
-stxdw [r10-1640], r1
-ldxdw r1, [r10-2240]
-stxdw [r10-1648], r1
-ldxdw r1, [r10-1856]
-stxdw [r10-1656], r1
-ldxdw r1, [r10-2192]
-stxdw [r10-1664], r1
-ldxb r1, [r10-342]
-stxb [r10-1617], r1
-ldxw r1, [r10-346]
-stxw [r10-1621], r1
-ldxdw r1, [r10-2344]
-stxb [r10-1574], r1
-ldxdw r1, [r10-2336]
-stxb [r10-1575], r1
-ldxdw r1, [r10-2328]
-stxb [r10-1576], r1
-ldxdw r1, [r10-2320]
-stxdw [r10-1584], r1
-ldxdw r1, [r10-2312]
-stxdw [r10-1592], r1
-ldxdw r1, [r10-2288]
-stxdw [r10-1600], r1
-ldxdw r1, [r10-1872]
-stxdw [r10-1608], r1
-ldxdw r1, [r10-2296]
-stxdw [r10-1616], r1
-ldxb r1, [r10-347]
-stxb [r10-1569], r1
-ldxw r1, [r10-351]
-stxw [r10-1573], r1
-ldxdw r1, [r10-2392]
-stxb [r10-1526], r1
-ldxdw r1, [r10-2384]
-stxb [r10-1527], r1
-ldxdw r1, [r10-2376]
-stxb [r10-1528], r1
-ldxdw r1, [r10-2368]
-stxdw [r10-1536], r1
-ldxdw r1, [r10-2360]
-stxdw [r10-1544], r1
-ldxdw r1, [r10-2352]
-stxdw [r10-1552], r1
-ldxdw r1, [r10-1880]
-stxdw [r10-1560], r1
-ldxdw r1, [r10-2304]
-stxdw [r10-1568], r1
-ldxb r1, [r10-352]
-stxb [r10-1521], r1
-ldxw r1, [r10-356]
-stxw [r10-1525], r1
-ldxdw r1, [r10-2456]
-stxb [r10-1478], r1
-ldxdw r1, [r10-2448]
-stxb [r10-1479], r1
-ldxdw r1, [r10-2440]
-stxb [r10-1480], r1
-ldxdw r1, [r10-2432]
-stxdw [r10-1488], r1
-ldxdw r1, [r10-2424]
-stxdw [r10-1496], r1
-ldxdw r1, [r10-2400]
-stxdw [r10-1504], r1
-ldxdw r1, [r10-1888]
-stxdw [r10-1512], r1
-ldxdw r1, [r10-2408]
-stxdw [r10-1520], r1
-ldxb r1, [r10-357]
-stxb [r10-1473], r1
-ldxw r1, [r10-361]
-stxw [r10-1477], r1
-ldxdw r1, [r10-2504]
-stxb [r10-1430], r1
-ldxdw r1, [r10-2496]
-stxb [r10-1431], r1
-ldxdw r1, [r10-2488]
-stxb [r10-1432], r1
-ldxdw r1, [r10-2480]
-stxdw [r10-1440], r1
-ldxdw r1, [r10-2472]
-stxdw [r10-1448], r1
-ldxdw r1, [r10-2464]
-stxdw [r10-1456], r1
-ldxdw r1, [r10-1904]
-stxdw [r10-1464], r1
-ldxdw r1, [r10-2416]
-stxdw [r10-1472], r1
-ldxb r1, [r10-362]
-stxb [r10-1425], r1
-ldxw r1, [r10-366]
-stxw [r10-1429], r1
-ldxdw r1, [r10-2560]
-stxb [r10-1382], r1
-ldxdw r1, [r10-2552]
-stxb [r10-1383], r1
-ldxdw r1, [r10-2544]
-stxb [r10-1384], r1
-ldxdw r1, [r10-2536]
-stxdw [r10-1392], r1
-ldxdw r1, [r10-2528]
-stxdw [r10-1400], r1
-ldxdw r1, [r10-2512]
-stxdw [r10-1408], r1
-ldxdw r1, [r10-1920]
-stxdw [r10-1416], r1
-ldxdw r1, [r10-2624]
-stxdw [r10-1424], r1
-ldxb r1, [r10-367]
-stxb [r10-1377], r1
-ldxw r1, [r10-371]
-stxw [r10-1381], r1
-ldxdw r1, [r10-2608]
-stxb [r10-1334], r1
-ldxdw r1, [r10-2600]
-stxb [r10-1335], r1
-ldxdw r1, [r10-2592]
-stxb [r10-1336], r1
-ldxdw r1, [r10-2584]
-stxdw [r10-1344], r1
-ldxdw r1, [r10-2576]
-stxdw [r10-1352], r1
-ldxdw r1, [r10-2568]
-stxdw [r10-1360], r1
-ldxdw r1, [r10-1928]
-stxdw [r10-1368], r1
-ldxdw r1, [r10-2520]
-stxdw [r10-1376], r1
-ldxb r1, [r10-372]
-stxb [r10-1329], r1
-ldxw r1, [r10-376]
-stxw [r10-1333], r1
-ldxdw r1, [r10-2672]
-stxb [r10-1286], r1
-ldxdw r1, [r10-2664]
-stxb [r10-1287], r1
-ldxdw r1, [r10-2656]
-stxb [r10-1288], r1
-ldxdw r1, [r10-2648]
-stxdw [r10-1296], r1
-ldxdw r1, [r10-2640]
-stxdw [r10-1304], r1
-ldxdw r1, [r10-2616]
-stxdw [r10-1312], r1
-ldxdw r1, [r10-1936]
-stxdw [r10-1320], r1
-ldxdw r1, [r10-2848]
-stxdw [r10-1328], r1
-ldxb r1, [r10-377]
-stxb [r10-1281], r1
-ldxw r1, [r10-381]
-stxw [r10-1285], r1
-ldxdw r1, [r10-2728]
-stxb [r10-1238], r1
-ldxdw r1, [r10-2720]
-stxb [r10-1239], r1
-ldxdw r1, [r10-2712]
-stxb [r10-1240], r1
-ldxdw r1, [r10-2704]
-stxdw [r10-1248], r1
-ldxdw r1, [r10-2696]
-stxdw [r10-1256], r1
-ldxdw r1, [r10-2680]
-stxdw [r10-1264], r1
-ldxdw r1, [r10-1944]
-stxdw [r10-1272], r1
-ldxdw r1, [r10-2632]
-stxdw [r10-1280], r1
-ldxb r1, [r10-382]
-stxb [r10-1233], r1
-ldxw r1, [r10-386]
-stxw [r10-1237], r1
-ldxdw r1, [r10-2784]
-stxb [r10-1190], r1
-ldxdw r1, [r10-2776]
-stxb [r10-1191], r1
-ldxdw r1, [r10-2768]
-stxb [r10-1192], r1
-ldxdw r1, [r10-2760]
-stxdw [r10-1200], r1
-ldxdw r1, [r10-2752]
-stxdw [r10-1208], r1
-ldxdw r1, [r10-2736]
-stxdw [r10-1216], r1
-ldxdw r1, [r10-1952]
-stxdw [r10-1224], r1
-ldxdw r1, [r10-2688]
-stxdw [r10-1232], r1
-ldxb r1, [r10-387]
-stxb [r10-1185], r1
-ldxw r1, [r10-391]
-stxw [r10-1189], r1
-ldxdw r1, [r10-2832]
-stxb [r10-1142], r1
-ldxdw r1, [r10-2824]
-stxb [r10-1143], r1
-ldxdw r1, [r10-2816]
-stxb [r10-1144], r1
-ldxdw r1, [r10-2808]
-stxdw [r10-1152], r1
-ldxdw r1, [r10-2800]
-stxdw [r10-1160], r1
-ldxdw r1, [r10-2792]
-stxdw [r10-1168], r1
-ldxdw r1, [r10-1960]
-stxdw [r10-1176], r1
-ldxdw r1, [r10-2744]
-stxdw [r10-1184], r1
-ldxb r1, [r10-392]
-stxb [r10-1137], r1
-ldxw r1, [r10-396]
-stxw [r10-1141], r1
-ldxdw r1, [r10-2896]
-stxb [r10-1094], r1
-ldxdw r1, [r10-2888]
-stxb [r10-1095], r1
-ldxdw r1, [r10-2880]
-stxb [r10-1096], r1
-ldxdw r1, [r10-2872]
-stxdw [r10-1104], r1
-ldxdw r1, [r10-2864]
-stxdw [r10-1112], r1
-ldxdw r1, [r10-2840]
-stxdw [r10-1120], r1
-ldxdw r1, [r10-1976]
-stxdw [r10-1128], r1
-ldxdw r1, [r10-2960]
-stxdw [r10-1136], r1
-ldxb r1, [r10-397]
-stxb [r10-1089], r1
-ldxw r1, [r10-401]
-stxw [r10-1093], r1
-ldxdw r1, [r10-2944]
-stxb [r10-1046], r1
-ldxdw r1, [r10-2936]
-stxb [r10-1047], r1
-ldxdw r1, [r10-2928]
-stxb [r10-1048], r1
-ldxdw r1, [r10-2920]
-stxdw [r10-1056], r1
-ldxdw r1, [r10-2912]
-stxdw [r10-1064], r1
-ldxdw r1, [r10-2904]
-stxdw [r10-1072], r1
-ldxdw r1, [r10-1984]
-stxdw [r10-1080], r1
-ldxdw r1, [r10-2856]
-stxdw [r10-1088], r1
-ldxb r1, [r10-402]
-stxb [r10-1041], r1
-ldxw r1, [r10-406]
-stxw [r10-1045], r1
-ldxdw r1, [r10-3008]
-stxb [r10-998], r1
-ldxdw r1, [r10-3000]
-stxb [r10-999], r1
-ldxdw r1, [r10-2992]
-stxb [r10-1000], r1
-ldxdw r1, [r10-2984]
-stxdw [r10-1008], r1
-ldxdw r1, [r10-2976]
-stxdw [r10-1016], r1
-ldxdw r1, [r10-2952]
-stxdw [r10-1024], r1
-ldxdw r1, [r10-1992]
-stxdw [r10-1032], r1
-ldxdw r1, [r10-3024]
-stxdw [r10-1040], r1
-ldxb r1, [r10-407]
-stxb [r10-993], r1
-ldxw r1, [r10-411]
-stxw [r10-997], r1
-ldxdw r1, [r10-3064]
-stxb [r10-950], r1
-ldxdw r1, [r10-3056]
-stxb [r10-951], r1
-ldxdw r1, [r10-3048]
-stxb [r10-952], r1
-ldxdw r1, [r10-3040]
-stxdw [r10-960], r1
-ldxdw r1, [r10-3032]
-stxdw [r10-968], r1
-ldxdw r1, [r10-3016]
-stxdw [r10-976], r1
-ldxdw r1, [r10-2000]
-stxdw [r10-984], r1
-ldxdw r1, [r10-2968]
-stxdw [r10-992], r1
-ldxb r1, [r10-412]
-stxb [r10-945], r1
-ldxw r1, [r10-416]
-stxw [r10-949], r1
-ldxdw r1, [r10-3120]
-stxb [r10-902], r1
-ldxdw r1, [r10-3112]
-stxb [r10-903], r1
-ldxdw r1, [r10-3104]
-stxb [r10-904], r1
-ldxdw r1, [r10-3096]
-stxdw [r10-912], r1
-ldxdw r1, [r10-3088]
-stxdw [r10-920], r1
-ldxdw r1, [r10-3072]
-stxdw [r10-928], r1
-ldxdw r1, [r10-2008]
-stxdw [r10-936], r1
-ldxdw r1, [r10-3184]
-stxdw [r10-944], r1
-ldxb r1, [r10-417]
-stxb [r10-897], r1
-ldxw r1, [r10-421]
-stxw [r10-901], r1
-ldxdw r1, [r10-3168]
-stxb [r10-854], r1
-ldxdw r1, [r10-3160]
-stxb [r10-855], r1
-ldxdw r1, [r10-3152]
-stxb [r10-856], r1
-ldxdw r1, [r10-3144]
-stxdw [r10-864], r1
-ldxdw r1, [r10-3136]
-stxdw [r10-872], r1
-ldxdw r1, [r10-3128]
-stxdw [r10-880], r1
-ldxdw r1, [r10-2016]
-stxdw [r10-888], r1
-ldxdw r1, [r10-3080]
-stxdw [r10-896], r1
-ldxb r1, [r10-422]
-stxb [r10-849], r1
-ldxw r1, [r10-426]
-stxw [r10-853], r1
-ldxdw r1, [r10-3232]
-stxb [r10-806], r1
-ldxdw r1, [r10-3224]
-stxb [r10-807], r1
-ldxdw r1, [r10-3216]
-stxb [r10-808], r1
-ldxdw r1, [r10-3208]
-stxdw [r10-816], r1
-ldxdw r1, [r10-3200]
-stxdw [r10-824], r1
-ldxdw r1, [r10-3176]
-stxdw [r10-832], r1
-ldxdw r1, [r10-2024]
-stxdw [r10-840], r1
-ldxdw r1, [r10-3304]
-stxdw [r10-848], r1
-ldxb r1, [r10-427]
-stxb [r10-801], r1
-ldxw r1, [r10-431]
-stxw [r10-805], r1
-ldxdw r1, [r10-3280]
-stxb [r10-758], r1
-ldxdw r1, [r10-3272]
-stxb [r10-759], r1
-ldxdw r1, [r10-3264]
-stxb [r10-760], r1
-ldxdw r1, [r10-3256]
-stxdw [r10-768], r1
-ldxdw r1, [r10-3248]
-stxdw [r10-776], r1
-ldxdw r1, [r10-3240]
-stxdw [r10-784], r1
-ldxdw r1, [r10-2032]
-stxdw [r10-792], r1
-ldxdw r1, [r10-3192]
-stxdw [r10-800], r1
-ldxb r1, [r10-432]
-stxb [r10-753], r1
-ldxw r1, [r10-436]
-stxw [r10-757], r1
-ldxdw r1, [r10-3352]
-stxb [r10-710], r1
-ldxdw r1, [r10-3344]
-stxb [r10-711], r1
-ldxdw r1, [r10-3336]
-stxb [r10-712], r1
-ldxdw r1, [r10-3328]
-stxdw [r10-720], r1
-ldxdw r1, [r10-3320]
-stxdw [r10-728], r1
-ldxdw r1, [r10-3296]
-stxdw [r10-736], r1
-ldxdw r1, [r10-2040]
-stxdw [r10-744], r1
-ldxdw r1, [r10-3360]
-stxdw [r10-752], r1
-ldxb r1, [r10-437]
-stxb [r10-705], r1
-ldxw r1, [r10-441]
-stxw [r10-709], r1
-ldxdw r1, [r10-3408]
-stxb [r10-662], r1
-ldxdw r1, [r10-3400]
-stxb [r10-663], r1
-ldxdw r1, [r10-3392]
-stxb [r10-664], r1
-ldxdw r1, [r10-3384]
-stxdw [r10-672], r1
-ldxdw r1, [r10-3416]
-stxdw [r10-680], r1
-ldxdw r1, [r10-2048]
-stxdw [r10-688], r1
-ldxdw r1, [r10-3312]
-stxdw [r10-696], r1
-ldxdw r1, [r10-3368]
-stxdw [r10-704], r1
-ldxb r1, [r10-442]
-stxb [r10-657], r1
-ldxw r1, [r10-446]
-stxw [r10-661], r1
-ldxdw r1, [r10-3464]
-stxb [r10-614], r1
-ldxdw r1, [r10-3456]
-stxb [r10-615], r1
-ldxdw r1, [r10-3448]
-stxb [r10-616], r1
-ldxdw r1, [r10-3440]
-stxdw [r10-624], r1
-ldxdw r1, [r10-3432]
-stxdw [r10-632], r1
-stxdw [r10-640], r0
-ldxdw r1, [r10-3376]
-stxdw [r10-648], r1
-ldxdw r1, [r10-3472]
-stxdw [r10-656], r1
-ldxb r1, [r10-447]
-stxb [r10-609], r1
-ldxw r1, [r10-451]
-stxw [r10-613], r1
-ldxdw r1, [r10-3520]
-stxb [r10-566], r1
-ldxdw r1, [r10-3512]
-stxb [r10-567], r1
-ldxdw r1, [r10-3504]
-stxb [r10-568], r1
-ldxdw r1, [r10-3496]
-stxdw [r10-576], r1
-ldxdw r1, [r10-3488]
-stxdw [r10-584], r1
-ldxdw r1, [r10-2056]
-stxdw [r10-592], r1
-stxdw [r10-600], r6
-ldxdw r1, [r10-3424]
-stxdw [r10-608], r1
-ldxb r1, [r10-452]
-stxb [r10-561], r1
-ldxw r1, [r10-456]
-stxw [r10-565], r1
-ldxdw r1, [r10-3544]
-stxb [r10-518], r1
-ldxdw r1, [r10-3536]
-stxb [r10-519], r1
-ldxdw r1, [r10-3528]
-stxb [r10-520], r1
-stxdw [r10-528], r8
-stxdw [r10-536], r5
-stxdw [r10-544], r9
-stxdw [r10-552], r7
-ldxdw r1, [r10-3480]
-stxdw [r10-560], r1
-ldxb r1, [r10-457]
-stxb [r10-513], r1
-ldxw r1, [r10-461]
-stxw [r10-517], r1
-stdw [r10-1816], 0
-stdw [r10-1824], 8
-stdw [r10-1832], 0
-ldxdw r1, [r10-24]
-stxdw [r10-472], r1
-ldxdw r1, [r10-32]
-stxdw [r10-480], r1
-ldxdw r1, [r10-40]
-stxdw [r10-488], r1
-ldxdw r1, [r10-48]
-stxdw [r10-496], r1
-mov64 r1, r10
-add64 r1, -336
-stxdw [r10-512], r1
-stdw [r10-504], 1
-mov64 r1, r10
-add64 r1, -1832
-ldxdw r2, [r10-1968]
-ldxdw r3, [r10-2072]
-mov64 r4, 0
-call pump_cpi::pump_v2::pump::cpi::buy_v2#6557
-mov64 r7, r0
-mov64 r6, r1
-jne r7, 2, +0x3a70
-ldxdw r9, [r10-1864]
-ja +0x3c18
-mov64 r1, r10
-add64 r1, -312
-lddw r2, .rodata[08000000]
-call mayhem_program::instructions::pump_amm_interface::pump_swap_types::PumpSwapError::name#fdf6
-stdw [r10-264], 0
-stdw [r10-272], 1
-stdw [r10-280], 0
-lddw r1, table[0:core::ptr::drop_in_place<alloc::string::String>#91b5, 3:<alloc::string::String as core::fmt::Write>::write_str#ca2d, 4:<alloc::string::String as core::fmt::Write>::write_char#6537, 5:core::fmt::Write::write_fmt#a290]
-stxdw [r10-40], r1
-mov64 r1, r10
-add64 r1, -280
-stxdw [r10-48], r1
-lddw r1, 0xe0000020
-stxdw [r10-32], r1
-mov64 r2, r10
-add64 r2, -48
-lddw r1, .rodata[08000000]
-call <mayhem_program::instructions::pump_amm_interface::pump_swap_types::PumpSwapError as core::fmt::Display>::fmt#0036
-ldxdw r9, [r10-1864]
-jne r0, 0, +0x47a0
-ldxdw r1, [r10-264]
-stxdw [r10-1760], r1
-ldxdw r1, [r10-272]
-stxdw [r10-1768], r1
-ldxdw r1, [r10-280]
-stxdw [r10-1776], r1
-ldxdw r1, [r10-312]
-stxdw [r10-1800], r1
-ldxdw r1, [r10-304]
-stxdw [r10-1792], r1
-ldxdw r1, [r10-296]
-stxdw [r10-1784], r1
-lddw r1, .rodata[70726f6772616d73]
-stxdw [r10-1824], r1
-stw [r10-1680], 6008
-stb [r10-1752], 2
-stw [r10-1808], 376
-stdw [r10-1816], 61
-stdw [r10-1832], 0
-mov64 r1, r10
-add64 r1, -1832
-call <anchor_lang::error::Error as core::convert::From<anchor_lang::error::AnchorError>>::from#5ca3
-jeq r7, 0, +0x3c08
-ldxdw r2, [r6]
-ldxb r2, [r6+40]
-ja +0x3c10
-ldxdw r2, [r6]
-jne r0, 2, +0x4b0
-ldxdw r1, [r9+48]
-call anchor_lang::accounts::interface_account::InterfaceAccount<T>::reload#d655
-jne r0, 2, +0x4b0
-ldxdw r1, [r9+104]
-call anchor_lang::accounts::account::Account<T>::reload#fa41
-jne r0, 2, +0x4b0
-ldxdw r1, [r10-3568]
-jeq r1, 0, +0x3c70
-ldxdw r1, [r9+64]
-call solana_account_info::AccountInfo::lamports#aa02
-ja +0x3c98
-ldxdw r1, [r9+56]
-call anchor_lang::accounts::interface_account::InterfaceAccount<T>::reload#d655
-jne r0, 2, +0x4b0
-ldxdw r1, [r9+56]
-ldxdw r0, [r1+104]
-ldxdw r1, [r10-3560]
-ldxdw r2, [r10-3552]
-jlt r1, r0, +0x48a8
-ldxdw r1, [r9+48]
-ldxdw r6, [r1+104]
-jlt r6, r2, +0x48c0
-ldxdw r3, [r10-3560]
-sub64 r3, r0
-sub64 r6, r2
-ldxdw r2, [r9+16]
-ldxdw r1, [r9+8]
-ldxdw r4, [r9+32]
-ldxdw r5, [r4]
-ldxdw r4, [r5+24]
-stxdw [r10-1808], r4
-ldxdw r4, [r5+16]
-stxdw [r10-1816], r4
-ldxdw r4, [r5+8]
-stxdw [r10-1824], r4
-ldxdw r4, [r5]
-stxdw [r10-1832], r4
-add64 r1, 8
-add64 r2, 8
-mov64 r5, r10
-add64 r5, -1832
-stxdw [r10-3560], r3
-mov64 r4, r6
-call orig_fn_2c478
-jne r0, 2, +0x4b0
-ldxdw r1, [r10-3568]
-jeq r1, 0, +0x3ea0
-ldxdw r1, [r9+64]
-call solana_account_info::AccountInfo::lamports#aa02
-stxdw [r10-1848], r0
-ldxdw r1, [r9+8]
-ldxdw r2, [r1+80]
-stxdw [r10-1840], r2
-ldxdw r4, [r1+72]
-ldxdw r1, [r9+16]
-ldxdw r0, [r1+48]
-ldxdw r8, [r1+40]
-ldxdw r5, [r9+104]
-ldxdw r7, [r5+96]
-ldxdw r2, [r5+80]
-stdw [r10-288], 0
-stdw [r10-296], 0
-stdw [r10-304], 0
-stdw [r10-312], 0
-ldxdw r3, [r10-3560]
-stxdw [r10-1960], r3
-stxdw [r10-1944], r7
-stxdw [r10-1880], r7
-stxdw [r10-1920], r4
-stxdw [r10-1856], r4
-ldxdw r3, [r10-1840]
-stxdw [r10-1864], r3
-stxdw [r10-1952], r2
-stxdw [r10-1888], r2
-stxdw [r10-1936], r8
-stxdw [r10-1904], r8
-stxdw [r10-1928], r0
-stxdw [r10-1912], r0
-ldxdw r2, [r10-1848]
-stxdw [r10-1872], r2
-ja +0x3fd0
-ldxdw r1, [r9+8]
-ldxdw r2, [r1+96]
-stxdw [r10-1864], r2
-ldxdw r1, [r1+88]
-stxdw [r10-1856], r1
-ldxdw r1, [r9+56]
-ldxdw r1, [r1+104]
-stxdw [r10-1872], r1
-ldxdw r1, [r9+16]
-ldxdw r2, [r1+48]
-stxdw [r10-1912], r2
-ldxdw r2, [r1+40]
-stxdw [r10-1904], r2
-ldxdw r5, [r9+104]
-ldxdw r3, [r5+96]
-stxdw [r10-1880], r3
-ldxdw r3, [r5+80]
-stxdw [r10-1888], r3
-ldxdw r3, [r9+32]
-ldxdw r3, [r3]
-ldxdw r4, [r3+24]
-stxdw [r10-288], r4
-ldxdw r4, [r3+16]
-stxdw [r10-296], r4
-ldxdw r4, [r3+8]
-stxdw [r10-304], r4
-ldxdw r3, [r3]
-stxdw [r10-312], r3
-mov64 r3, 0
-stxdw [r10-1944], r3
-stxdw [r10-1920], r3
-stxdw [r10-1840], r3
-stxdw [r10-1952], r3
-stxdw [r10-1936], r3
-stxdw [r10-1928], r3
-stxdw [r10-1848], r3
-mov64 r2, 0
-stxdw [r10-1960], r2
-ldxdw r3, [r9+64]
-ldxdw r3, [r3]
-ldxdw r4, [r3+24]
-stxdw [r10-256], r4
-ldxdw r4, [r3+16]
-stxdw [r10-264], r4
-ldxdw r4, [r3+8]
-stxdw [r10-272], r4
-ldxdw r3, [r3]
-stxdw [r10-280], r3
-ldxdw r3, [r9+24]
+stxdw [r10-288], r3
+ldxdw r3, [r8+24]
 ldxdw r3, [r3+88]
 ldxdw r3, [r3]
 ldxdw r4, [r3+24]
-stxdw [r10-24], r4
+stxdw [r10-32], r4
 ldxdw r4, [r3+16]
-stxdw [r10-32], r4
+stxdw [r10-40], r4
 ldxdw r4, [r3+8]
-stxdw [r10-40], r4
+stxdw [r10-48], r4
 ldxdw r3, [r3]
-stxdw [r10-48], r3
-ldxdw r2, [r1+64]
-stxdw [r10-1968], r2
-ldxdw r2, [r1+56]
-stxdw [r10-1976], r2
-ldxdw r1, [r1+80]
-stxdw [r10-1984], r1
-ldxdw r8, [r5+72]
-ldxdw r1, [r9+48]
-ldxdw r7, [r1+104]
-mov64 r1, r10
-add64 r1, -128
+stxdw [r10-56], r3
+jeq r7, 0, +0x1a88
+call solana_account_info::AccountInfo::lamports#aa02
+stxdw [r10-3376], r0
+ldxdw r1, [r8+8]
+ldxdw r2, [r1+80]
+stxdw [r10-3344], r2
+ldxdw r1, [r1+72]
+stxdw [r10-3360], r1
+ldxdw r2, [r8+16]
+ldxdw r1, [r2+48]
+stxdw [r10-3392], r1
+ldxdw r1, [r2+40]
+stxdw [r10-3384], r1
+ldxdw r1, [r8+104]
+ldxdw r3, [r1+80]
+stxdw [r10-3408], r3
+ldxdw r3, [r10-3352]
+stxdw [r10-3400], r3
+ja +0x1ad8
+mov64 r1, r10
+add64 r1, -552
+lddw r2, .rodata[06000000]
+call mayhem_program::instructions::pump_interface::buy::BuyError::name#3012
+mov64 r1, r10
+add64 r1, -528
+mov64 r2, 6
+call <T as alloc::string::SpecToString>::spec_to_string#f398
+lddw r1, .rodata[70726f6772616d73]
+stxdw [r10-576], r1
+stb [r10-504], 2
+stw [r10-560], 115
+stdw [r10-568], 61
+stdw [r10-584], 0
+stw [r10-432], 6006
+mov64 r1, r10
+add64 r1, -584
+call <anchor_lang::error::Error as core::convert::From<anchor_lang::error::AnchorError>>::from#5ca3
+mov64 r2, r1
+mov64 r1, r0
+mov64 r3, r8
+mov64 r4, r9
+call anchor_lang::error::Error::with_values#4132
+ja +0x5a0
+mov64 r1, 0
+stxdw [r10-3400], r1
+ldxdw r1, [r8+104]
+mov64 r3, 0
+stxdw [r10-3376], r3
+stxdw [r10-3384], r3
+stxdw [r10-3392], r3
+stxdw [r10-3408], r3
+stxdw [r10-3360], r3
+stxdw [r10-3344], r3
+ldxdw r6, [r2+64]
+ldxdw r2, [r2+56]
+stxdw [r10-3424], r2
+ldxdw r1, [r1+72]
+stxdw [r10-3416], r1
+ldxdw r1, [r8+48]
+ldxdw r9, [r1+104]
+mov64 r1, r10
+add64 r1, -136
 call solana_sysvar::clock::<impl solana_sysvar::Sysvar for solana_clock::Clock>::get#06ec
-ldxw r1, [r10-128]
-jne r1, 1, +0x4108
-ldxw r2, [r10-120]
-ldxw r1, [r10-124]
+ldxw r1, [r10-136]
+jne r1, 1, +0x1b58
+ldxw r2, [r10-128]
+ldxw r1, [r10-132]
 call <anchor_lang::error::Error as core::convert::From<solana_program_error::ProgramError>>::from#507a
-ja +0x4b0
-ldxdw r2, [r10-88]
-ldxdw r1, [r9+8]
-ldxw r1, [r1+248]
-ldxdw r3, [r9+104]
-ldxdw r3, [r3+88]
-ldxdw r4, [r10-288]
-stxdw [r10-1552], r4
-ldxdw r4, [r10-296]
-stxdw [r10-1560], r4
-ldxdw r4, [r10-304]
-stxdw [r10-1568], r4
-ldxdw r4, [r10-312]
-stxdw [r10-1576], r4
-ldxdw r4, [r10-280]
-stxdw [r10-1832], r4
-ldxdw r4, [r10-272]
-stxdw [r10-1824], r4
-ldxdw r4, [r10-264]
-stxdw [r10-1816], r4
-ldxdw r4, [r10-256]
-stxdw [r10-1808], r4
-stb [r10-1538], 0
-ldxdw r4, [r10-1960]
-stxdw [r10-1768], r4
-stxdw [r10-1760], r6
-ldxdw r4, [r10-1848]
-stxdw [r10-1752], r4
-stxdw [r10-1744], r7
-ldxdw r4, [r10-1952]
-stxdw [r10-1704], r4
-stxdw [r10-1696], r8
-stxdw [r10-1672], r2
-ldxdw r2, [r10-1984]
-stxdw [r10-1664], r2
-ldxdw r2, [r10-1944]
-stxdw [r10-1656], r2
-stxdw [r10-1648], r3
-ldxdw r2, [r10-1896]
-stxh [r10-1540], r2
-stxw [r10-1544], r1
-ldxdw r1, [r10-1920]
-stxdw [r10-1688], r1
-ldxdw r1, [r10-1840]
-stxdw [r10-1680], r1
-ldxdw r1, [r10-1976]
-stxdw [r10-1720], r1
-ldxdw r1, [r10-1968]
-stxdw [r10-1712], r1
-ldxdw r1, [r10-1936]
-stxdw [r10-1736], r1
-ldxdw r1, [r10-1928]
-stxdw [r10-1728], r1
-ldxdw r1, [r10-24]
-stxdw [r10-1776], r1
-ldxdw r1, [r10-32]
-stxdw [r10-1784], r1
-ldxdw r1, [r10-40]
-stxdw [r10-1792], r1
-ldxdw r1, [r10-48]
-stxdw [r10-1800], r1
-ldxdw r1, [r10-1864]
-stxdw [r10-1592], r1
-ldxdw r1, [r10-1856]
-stxdw [r10-1600], r1
-ldxdw r1, [r10-1912]
-stxdw [r10-1616], r1
-ldxdw r1, [r10-1904]
-stxdw [r10-1624], r1
-ldxdw r1, [r10-1880]
-stxdw [r10-1584], r1
-ldxdw r1, [r10-1888]
-stxdw [r10-1608], r1
-ldxdw r1, [r10-1872]
-stxdw [r10-1632], r1
-ldxdw r1, [r10-3560]
-stxdw [r10-1640], r1
-mov64 r1, r10
-add64 r1, -128
+ja +0x5a0
+ldxdw r3, [r8+16]
+ldxdw r0, [r8+104]
+ldxdw r2, [r8+8]
+mov64 r5, r2
+ldxdw r1, [r10-3328]
+add64 r5, r1
+ldxdw r1, [r0+96]
+mov64 r4, r1
+jne r7, 0, +0x1ba8
+mov64 r4, 0
+ldxdw r8, [r10-96]
+stxdw [r10-3464], r8
+ldxdw r8, [r3+48]
+stxdw [r10-3432], r8
+ldxdw r8, [r3+40]
+stxdw [r10-3328], r8
+ldxdw r8, [r0+80]
+ldxw r2, [r2+224]
+stxdw [r10-3456], r2
+ldxdw r0, [r0+88]
+ldxdw r3, [r3+80]
+ldxdw r2, [r5+8]
+stxdw [r10-3448], r2
+ldxdw r2, [r5]
+stxdw [r10-3440], r2
+ldxdw r5, [r10-3320]
+jeq r7, 0, +0x1c58
+stdw [r10-112], 0
+stdw [r10-120], 0
+stdw [r10-128], 0
+stdw [r10-136], 0
+ja +0x1cb0
+ldxdw r2, [r10-3368]
+ldxdw r2, [r2+32]
+ldxdw r2, [r2]
+ldxdw r7, [r2+24]
+stxdw [r10-112], r7
+ldxdw r7, [r2+16]
+stxdw [r10-120], r7
+ldxdw r7, [r2+8]
+stxdw [r10-128], r7
+ldxdw r2, [r2]
+stxdw [r10-136], r2
+ldxdw r7, [r10-3344]
+ldxdw r2, [r10-288]
+stxdw [r10-584], r2
+ldxdw r2, [r10-280]
+stxdw [r10-576], r2
+ldxdw r2, [r10-272]
+stxdw [r10-568], r2
+ldxdw r2, [r10-264]
+stxdw [r10-560], r2
+ldxdw r2, [r10-56]
+stxdw [r10-552], r2
+ldxdw r2, [r10-48]
+stxdw [r10-544], r2
+ldxdw r2, [r10-40]
+stxdw [r10-536], r2
+ldxdw r2, [r10-32]
+stxdw [r10-528], r2
+stxdw [r10-384], r9
+ldxdw r2, [r10-3376]
+stxdw [r10-392], r2
+ldxdw r2, [r10-3312]
+stxdw [r10-400], r2
+ldxdw r2, [r10-3400]
+stxdw [r10-408], r2
+stb [r10-290], 0
+stxdw [r10-464], r6
+ldxdw r2, [r10-3424]
+stxdw [r10-472], r2
+ldxdw r2, [r10-3392]
+stxdw [r10-480], r2
+ldxdw r2, [r10-3384]
+stxdw [r10-488], r2
+ldxdw r2, [r10-3416]
+stxdw [r10-368], r2
+ldxdw r2, [r10-3408]
+stxdw [r10-376], r2
+ldxdw r2, [r10-3464]
+stxdw [r10-360], r2
+stxdw [r10-352], r3
+stxdw [r10-344], r4
+stxdw [r10-336], r0
+ldxdw r2, [r10-3336]
+stxh [r10-292], r2
+ldxdw r2, [r10-3456]
+stxw [r10-296], r2
+ldxdw r2, [r10-3360]
+stxdw [r10-456], r2
+stxdw [r10-448], r7
+ldxdw r2, [r10-3352]
+stxdw [r10-328], r2
+stxdw [r10-320], r5
+stxdw [r10-312], r8
+ldxdw r2, [r10-3328]
+stxdw [r10-440], r2
+ldxdw r2, [r10-3432]
+stxdw [r10-432], r2
+ldxdw r2, [r10-3448]
+stxdw [r10-416], r2
+ldxdw r2, [r10-3440]
+stxdw [r10-424], r2
+stxdw [r10-304], r1
+ldxdw r1, [r10-112]
+stxdw [r10-496], r1
+ldxdw r1, [r10-120]
+stxdw [r10-504], r1
+ldxdw r1, [r10-128]
+stxdw [r10-512], r1
+ldxdw r1, [r10-136]
+stxdw [r10-520], r1
+mov64 r1, r10
+add64 r1, -136
 mov64 r6, r10
-add64 r6, -1832
+add64 r6, -584
 mov64 r2, r6
 call <mayhem_program::instructions::pump_interface::sell::SellEvent as anchor_lang::Event>::data#c4f7
-ldxdw r1, [r10-112]
-stxdw [r10-40], r1
 ldxdw r1, [r10-120]
 stxdw [r10-48], r1
-mov64 r1, r10
-add64 r1, -48
+ldxdw r1, [r10-128]
+stxdw [r10-56], r1
+mov64 r1, r10
+add64 r1, -56
 mov64 r8, 1
 mov64 r2, 1
 call anchor_lang::solana_program::log::sol_log_data#9398
-ldxdw r2, [r9+224]
-mov64 r1, r10
-add64 r1, -248
-call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
-mov64 r1, r10
-add64 r1, -200
+ldxdw r1, [r10-3368]
+ldxdw r2, [r1+224]
+mov64 r1, r10
+add64 r1, -256
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+mov64 r1, r10
+add64 r1, -208
 mov64 r2, r6
 call <mayhem_program::instructions::pump_interface::sell::SellEvent as anchor_lang::Event>::data#c4f7
 lddw r1, .rodata[e445a52e51cb9a1d]
 lddw r2, .rodata[e445a52e51cb9a1d]
 add64 r2, 8
-stxdw [r10-136], r2
-stxdw [r10-144], r1
+stxdw [r10-144], r2
+stxdw [r10-152], r1
+ldxdw r1, [r10-208]
+stxdw [r10-168], r1
 ldxdw r1, [r10-200]
+stxdw [r10-176], r1
+stxdw [r10-184], r1
+ldxdw r2, [r10-192]
+add64 r1, r2
 stxdw [r10-160], r1
-ldxdw r1, [r10-192]
-stxdw [r10-168], r1
-stxdw [r10-176], r1
-ldxdw r2, [r10-184]
-add64 r1, r2
-stxdw [r10-152], r1
-mov64 r1, r10
-add64 r1, -280
+mov64 r1, r10
+add64 r1, -288
 mov64 r2, r10
-add64 r2, -176
+add64 r2, -184
 call <alloc::vec::Vec<T> as alloc::vec::spec_from_iter_nested::SpecFromIterNested<T,I>>::from_iter#1a04
-ldxdw r6, [r10-272]
-ldxdw r7, [r10-264]
+ldxdw r6, [r10-280]
+ldxdw r7, [r10-272]
 call _RNvCscR3e5J6VSLN_7___rustc35___rust_no_alloc_shim_is_unstable_v2
 lddw r1, 0x300000000
 ldxdw r2, [r1]
 mov64 r3, 0
 mov64 r1, r2
 add64 r1, -34
-jgt r1, r2, +0x4570
+jgt r1, r2, +0x20e8
 mov64 r8, 0
-jeq r8, 0, +0x4578
+jeq r8, 0, +0x20f0
 lddw r1, 0x300007fde
-jeq r2, 0, +0x4598
+jeq r2, 0, +0x2110
 mov64 r1, r3
 lddw r2, 0x300000008
-jge r1, r2, +0x45b0
+jge r1, r2, +0x2128
 mov64 r1, 1
 mov64 r2, 34
 call alloc::alloc::handle_alloc_error#decd
-jne r8, 0, +0x4520
+jne r8, 0, +0x2098
 mov64 r3, r1
 lddw r1, 0x300007fde
-jne r2, 0, +0x4538
+jne r2, 0, +0x20b0
 lddw r2, 0x300000008
-jlt r1, r2, +0x4558
+jlt r1, r2, +0x20d0
 lddw r2, 0x300000000
 stxdw [r2], r1
-ldxdw r2, [r10-248]
+ldxdw r2, [r10-256]
 ldxdw r3, [r2]
 ldxdw r4, [r2+8]
 ldxdw r5, [r2+16]
@@ -2210,22 +1042,22 @@
 stxdw [r1+8], r4
 stxdw [r1], r3
 sth [r1+32], 1
-stxdw [r10-40], r1
-stdw [r10-32], 1
-stdw [r10-48], 1
+stxdw [r10-48], r1
+stdw [r10-40], 1
+stdw [r10-56], 1
 mov64 r8, r10
-add64 r8, -128
+add64 r8, -136
 mov64 r5, r10
-add64 r5, -48
+add64 r5, -56
 mov64 r1, r8
 lddw r2, .rodata[052ae5d7a7daa724]
 mov64 r3, r6
 mov64 r4, r7
 call solana_instruction::Instruction::new_with_bytes#33e2
 mov64 r6, r10
-add64 r6, -48
+add64 r6, -56
 mov64 r2, r10
-add64 r2, -248
+add64 r2, -256
 mov64 r1, r6
 mov64 r3, 48
 call memcpy
@@ -2239,48 +1071,58 @@
 mov64 r1, r0
 lsh64 r1, 32
 rsh64 r1, 32
-jne r1, 26, +0x4730
+jne r1, 26, +0x22a0
 mov64 r6, 2
-ja +0x4750
-call abort
+ja +0x22c0
 mov64 r1, r0
 call <anchor_lang::error::Error as core::convert::From<solana_program_error::ProgramError>>::from#507a
 mov64 r6, r0
 mov64 r7, r1
 mov64 r1, r10
-add64 r1, -48
+add64 r1, -56
 call core::ptr::drop_in_place<[solana_account_info::AccountInfo; 1]>#240e
 mov64 r0, r6
 mov64 r1, r7
-jne r0, 2, +0x4b0
+jne r0, 2, +0x5a0
 mov64 r0, 2
-ja +0x4b0
-mov64 r9, 2
-ja +0x358
+ja +0x5a0
+ldxdw r1, [r10-580]
+stxdw [r10-136], r1
 mov64 r3, r10
-add64 r3, -341
-lddw r1, str("a Display implementation returned an error unexpectedly")
-mov64 r2, 55
-lddw r4, table[3:<core::fmt::Error as core::fmt::Debug>::fmt#aaa5]
-lddw r5, panic_location(/home/runner/work/platform-tools/platform-tools/out/rust/library/alloc/src/string.rs:2830:14)
+add64 r3, -136
+lddw r1, str("called `Result::unwrap()` on an `Err` value")
+mov64 r2, 43
+lddw r4, table[3:<solana_program_error::ProgramError as core::fmt::Debug>::fmt#4fcd]
+lddw r5, panic_location(programs/mayhem-program/src/instructions/pump_interface_v2.rs:77:23)
+call core::result::unwrap_failed#5a05
+ldxdw r1, [r10-580]
+stxdw [r10-136], r1
+mov64 r3, r10
+add64 r3, -136
+lddw r1, str("called `Result::unwrap()` on an `Err` value")
+mov64 r2, 43
+lddw r4, table[3:<solana_program_error::ProgramError as core::fmt::Debug>::fmt#4fcd]
+lddw r5, panic_location(programs/mayhem-program/src/instructions/pump_interface_v2.rs:82:24)
 call core::result::unwrap_failed#5a05
 mov64 r3, r10
-add64 r3, -341
+add64 r3, -1
 lddw r1, str("called `Result::unwrap()` on an `Err` value")
 mov64 r2, 43
 lddw r4, table[3:<core::num::error::TryFromIntError as core::fmt::Debug>::fmt#ac91]
-lddw r5, panic_location(programs/mayhem-program/src/state.rs:350:43)
+lddw r5, panic_location(programs/mayhem-program/src/instructions/pump_interface_v2.rs:96:54)
 call core::result::unwrap_failed#5a05
+lddw r1, panic_location(programs/mayhem-program/src/instructions/pump_interface_v2.rs:100:77)
+call core::option::unwrap_failed#2d9b
 mov64 r3, r10
-add64 r3, -341
+add64 r3, -1
 lddw r1, str("called `Result::unwrap()` on an `Err` value")
 mov64 r2, 43
 lddw r4, table[3:<core::num::error::TryFromIntError as core::fmt::Debug>::fmt#ac91]
-lddw r5, panic_location(programs/mayhem-program/src/state.rs:379:34)
+lddw r5, panic_location(programs/mayhem-program/src/instructions/pump_interface_v2.rs:100:87)
 call core::result::unwrap_failed#5a05
-lddw r1, panic_location(programs/mayhem-program/src/state.rs:321:26)
+lddw r1, panic_location(programs/mayhem-program/src/instructions/pump_interface_v2.rs:94:15)
 call core::panicking::panic_const::panic_const_sub_overflow#7310
-lddw r1, panic_location(programs/mayhem-program/src/instructions/pump_interface_v2.rs:393:25)
+lddw r1, panic_location(programs/mayhem-program/src/instructions/pump_interface_v2.rs:108:10)
 call core::panicking::panic_const::panic_const_sub_overflow#7310
-lddw r1, panic_location(programs/mayhem-program/src/instructions/pump_interface_v2.rs:394:26)
+lddw r1, panic_location(programs/mayhem-program/src/instructions/pump_interface_v2.rs:108:44)
 call core::panicking::panic_const::panic_const_sub_overflow#7310
